<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Quick tutorials &mdash; KGGSEE  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/kggsee.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Detailed Document" href="detailed_document.html" />
    <link rel="prev" title="Setup" href="setup.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> KGGSEE
          </a>
              <div class="version">
                v1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setup</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Quick tutorials</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#gene-based-association-tests">1. Gene-based association tests</a></li>
<li class="toctree-l2"><a class="reference internal" href="#dese">2. DESE</a></li>
<li class="toctree-l2"><a class="reference internal" href="#emic">3. EMIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gene-based-heritability-estimation">4. Gene-based heritability estimation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="detailed_document.html">Detailed Document</a></li>
<li class="toctree-l1"><a class="reference internal" href="options.html">Options</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">KGGSEE</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Quick tutorials</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/quick_tutorials.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="quick-tutorials">
<span id="id1"></span><h1>Quick tutorials<a class="headerlink" href="#quick-tutorials" title="Permalink to this headline"></a></h1>
<p>We provide four quick tutorials; each shows a function of KGGSEE. In each tutorial, we provide the command line and a brief explanation of flags and output files. Please refer to <a class="reference internal" href="detailed_document.html#detailed-document"><span class="std std-ref">Detailed Document</span></a> and <a class="reference internal" href="options.html#options"><span class="std std-ref">Options</span></a> for details. The first tutorial (<a class="reference internal" href="#t1"><span class="std std-ref">1. GATES and ECS</span></a>) should be done first, then there is no need to do the following in order.</p>
<p>Make sure the KGGSEE Java archive <code class="docutils literal notranslate"><span class="pre">kggsee.jar</span></code>, the running resource data folder <code class="docutils literal notranslate"><span class="pre">resources/</span></code>, and the tutorial data folder <code class="docutils literal notranslate"><span class="pre">tutorials/</span></code> are under the same folder. For convenience, enter the <code class="docutils literal notranslate"><span class="pre">tutorials/</span></code> directory first.</p>
<section id="gene-based-association-tests">
<span id="t1"></span><h2>1. Gene-based association tests<a class="headerlink" href="#gene-based-association-tests" title="Permalink to this headline"></a></h2>
<p>GATES and ECS are two statistical methods combining the p-values of a group of SNPs into one p-value. This analysis inputs p-values of SNPs and outputs p-values of genes. The command is:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>java -Xmx4g -jar ../kggsee.jar <span class="se">\</span>
--sum-file scz_gwas_eur_chr1.tsv.gz <span class="se">\</span>
--vcf-ref 1kg_hg19_eur_chr1.vcf.gz <span class="se">\</span>
--keep-ref <span class="se">\</span>
--gene-assoc <span class="se">\</span>
--out t1
</pre></div>
</div>
<p><strong>Explanation of the flags and input files:</strong></p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 80%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--sum-file</span></code></p></td>
<td><p>Specifies a whitespace delimitated file of GWAS summary statistics. In this analysis, columns of SNP coordinates and p-values (CHR, BP, and P by default) are needed.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--vcf-ref</span></code></p></td>
<td><p>Specifies a VCF file of genotypes sampled from a reference population. These genotypes are used to estimate LD correlation coefficients among SNPs.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--keep-ref</span></code></p></td>
<td><p>Keep the parsed VCF file (KGGSEE object format) in a folder named <code class="docutils literal notranslate"><span class="pre">VCFRefhg19</span></code> under the output folder. KGGSEE will read these files in the following tutorials, which will be faster than parsing VCF files.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--gene-assoc</span></code></p></td>
<td><p>Triggers gene-based association tests.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--out</span></code></p></td>
<td><p>Specifies the prefix of output files.</p></td>
</tr>
</tbody>
</table>
<p><strong>Explanation of the output files:</strong></p>
<p>The numeric results of gene-based association tests are saved in <code class="docutils literal notranslate"><span class="pre">t1.gene.pvalue.txt</span></code>. There are seven columns in the file:</p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 80%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Header</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Gene</p></td>
<td><p>Gene symbol</p></td>
</tr>
<tr class="row-odd"><td><p>#Var</p></td>
<td><p>Number of variants within the gene</p></td>
</tr>
<tr class="row-even"><td><p>ECSP</p></td>
<td><p>p-value of ECS</p></td>
</tr>
<tr class="row-odd"><td><p>GATESP</p></td>
<td><p>p-value of GATES</p></td>
</tr>
<tr class="row-even"><td><p>Chrom</p></td>
<td><p>Chromosome of the gene</p></td>
</tr>
<tr class="row-odd"><td><p>Pos</p></td>
<td><p>Coordinate of the variant with the lowest p-value within the gene</p></td>
</tr>
<tr class="row-even"><td><p>GWAS_Var_P</p></td>
<td><p>p-value of the variant</p></td>
</tr>
</tbody>
</table>
<p>The columns of <code class="docutils literal notranslate"><span class="pre">t1.gene.var.pvalue.txt.gz</span></code> are the same as <code class="docutils literal notranslate"><span class="pre">t1.gene.pvalue.txt</span></code>. The difference is that, for each gene, in <code class="docutils literal notranslate"><span class="pre">t1.gene.pvalue.txt</span></code>, only the variant with the lowest p-value is output, while in <code class="docutils literal notranslate"><span class="pre">t1.gene.var.pvalue.txt.gz</span></code>, all variants are output.</p>
<p>The Q-Q plots for p-values of inputted GWAS file (inside or outside of gene) and gene-based association tests by GATES or ECS are saved in <code class="docutils literal notranslate"><span class="pre">t1.qq.png</span></code>.</p>
</section>
<section id="dese">
<span id="t2"></span><h2>2. DESE<a class="headerlink" href="#dese" title="Permalink to this headline"></a></h2>
<p>DESE performs phenotype-tissue association tests and conditional gene-based association tests at the same time. This analysis inputs p-values of a GWAS and expression profile of multiple tissues; outputs p-values of phenotype-tissue associations and conditional p-values of genes. The command is:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>java -Xmx4g -jar ../kggsee.jar <span class="se">\</span>
--sum-file scz_gwas_eur_chr1.tsv.gz <span class="se">\</span>
--saved-ref VCFRefhg19 <span class="se">\</span>
--expression-file GTEx_v8_gene_meanSE.tsv.gz <span class="se">\</span>
--gene-finemapping <span class="se">\</span>
--out t2
</pre></div>
</div>
<p><strong>Explanation of the flags and input files:</strong></p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 75%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--sum-file</span></code></p></td>
<td><p>Specifies a whitespace delimitated file of GWAS summary statistics. In this analysis, columns of SNP coordinates and p-values are needed.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--saved-ref</span></code></p></td>
<td><p>Specifies the folder of genotypes of reference population in KGGSEE object format, which is saved by the <code class="docutils literal notranslate"><span class="pre">--keep-ref</span></code> flag in the first tutorial.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--expression-file</span></code></p></td>
<td><p>Specifies a gene expression file that contains means and standard errors of gene expressions in multiple tissues. Here <code class="docutils literal notranslate"><span class="pre">GTEx_v8_gene_meanSE.tsv.gz</span></code> is for gene-leveled DESE. Try <code class="docutils literal notranslate"><span class="pre">GTEx_v8_transcript_meanSE.tsv.gz</span></code> for transcript-leveled DESE.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--gene-finemapping</span></code></p></td>
<td><p>Triggers the DESE analysis.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--out</span></code></p></td>
<td><p>Specifies the prefix of output files.</p></td>
</tr>
</tbody>
</table>
<p><strong>Explanation of the output files:</strong></p>
<p>The three files of <code class="docutils literal notranslate"><span class="pre">t2.gene.pvalue.txt</span></code>, <code class="docutils literal notranslate"><span class="pre">t2.gene.var.pvalue.txt.gz</span></code>, and <code class="docutils literal notranslate"><span class="pre">t2.qq.png</span></code> are the same as their counterparts with the same suffixes of the first tutorial. In addition, the results of conditional gene-based association tests are in <code class="docutils literal notranslate"><span class="pre">t2.finemapping.gene.ecs.txt</span></code> which contains nine columns:</p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 80%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Header</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Gene</p></td>
<td><p>Gene symbol</p></td>
</tr>
<tr class="row-odd"><td><p>Chrom</p></td>
<td><p>Chromosome of the gene</p></td>
</tr>
<tr class="row-even"><td><p>StartPos</p></td>
<td><p>Start coordinate of the gene</p></td>
</tr>
<tr class="row-odd"><td><p>EndPos</p></td>
<td><p>End coordinate of the gene</p></td>
</tr>
<tr class="row-even"><td><p>#Var</p></td>
<td><p>Number of variants within the gene</p></td>
</tr>
<tr class="row-odd"><td><p>Group</p></td>
<td><p>LD group number. Conditional ECS tests were performed for genes within a same LD group.</p></td>
</tr>
<tr class="row-even"><td><p>ECSP</p></td>
<td><p>p-value of ECS</p></td>
</tr>
<tr class="row-odd"><td><p>CondiECSP</p></td>
<td><p>p-value of conditional gene-based association tests by conditional ECS</p></td>
</tr>
<tr class="row-even"><td><p>GeneScore</p></td>
<td><p>The gene’s selective expression score in all tissues. A gene with a high socre will be given higher priority to enter the conditioning procedure.</p></td>
</tr>
</tbody>
</table>
<p>Results of phenotype-tissue associations are in <code class="docutils literal notranslate"><span class="pre">t2.celltype.txt</span></code>. This is basically a Wilcoxon rank-sum test which tests whether the selective expression median of the phenotype-associated genes is significantly higher than that of other genes in an interrogated tissue. The file contains three columns:</p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 80%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Header</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>TissueName</p></td>
<td><p>Name of the tissue being tested</p></td>
</tr>
<tr class="row-odd"><td><p>p</p></td>
<td><p>p-values of phenotype-tissue associations.</p></td>
</tr>
<tr class="row-even"><td><p>Log(p)</p></td>
<td><p>The negative logarithm (base 10) of p-values of phenotype-tissue associations</p></td>
</tr>
</tbody>
</table>
</section>
<section id="emic">
<span id="t3"></span><h2>3. EMIC<a class="headerlink" href="#emic" title="Permalink to this headline"></a></h2>
<p>EMIC inferences gene expressions’ causal effect on a complex phenotype with dependent expression quantitative loci by a robust median-based Mendelian randomization. SNPs with effects on both the phenotype and a gene are considered instrumental variables (IVs) of the gene, which can be used to infer the gene’s expression effect on the phenotype. This analysis inputs effect sizes of SNPs on the phenotype and genes’ expressions; outputs effect sizes and p-values of genes’ expression effects on the phenotype. The command is:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>java -Xmx4g -jar ../kggsee.jar <span class="se">\</span>
--sum-file scz_gwas_eur_chr1.tsv.gz <span class="se">\</span>
--saved-ref VCFRefhg19 <span class="se">\</span>
--eqtl-file GTEx_v8_gene_BrainBA9.eqtl.txt.gz <span class="se">\</span>
--beta-col OR <span class="se">\</span>
--beta-type <span class="m">2</span> <span class="se">\</span>
--emic <span class="se">\</span>
--out t3
</pre></div>
</div>
<p><strong>Explanation of the flags and input files:</strong></p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 80%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Header</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Flag</p></td>
<td><p>Description</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--sum-file</span></code></p></td>
<td><p>Specifies a whitespace delimitated file of GWAS summary statistics. In this analysis, in addition to the columns of SNP coordinates and p-values, two columns of SNP alleles (named A1 and A2 by default), a column of A1 allele frequency (named FRQ_U by default), and two columns of SNP effect sizes (no default header) and their standard errors (named SE by default) are also needed.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--saved-ref</span></code></p></td>
<td><p>Specifies the folder of genotypes of reference population in KGGSEE object format, which is saved by the <code class="docutils literal notranslate"><span class="pre">--keep-ref</span></code> flag in the first tutorial.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--eqtl-file</span></code></p></td>
<td><p>Specifies a fasta-styled file of SNPs’ effects on gene expressions. Here <code class="docutils literal notranslate"><span class="pre">GTEx_v8_gene_BrainBA9.eqtl.txt.gz</span></code> for gene-leveled EMIC. Try <code class="docutils literal notranslate"><span class="pre">GTEx_v8_transcript_BrainBA9.eqtl.txt.gz</span></code> for transcript-leveled EMIC.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--beta-col</span></code></p></td>
<td><p>Specifies the column name of effect sizes in the GWAS file.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--beta-type</span></code></p></td>
<td><p>Specifies the type of the effect size; here <code class="docutils literal notranslate"><span class="pre">2</span></code> means that it is the odds ratio for a qualitative phenotype.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--emic</span></code></p></td>
<td><p>Triggers the EMIC analysis.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--out</span></code></p></td>
<td><p>Specifies the prefix of output files.</p></td>
</tr>
</tbody>
</table>
<p><strong>Explanation of the output files:</strong></p>
<p>The numeric results of EMIC are saved in <code class="docutils literal notranslate"><span class="pre">t3.emic.gene.txt</span></code>. There are nine columns in the file:</p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 80%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Header</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Gene</p></td>
<td><p>The gene symbol</p></td>
</tr>
<tr class="row-odd"><td><p>#Var</p></td>
<td><p>Number of IVs within the gene</p></td>
</tr>
<tr class="row-even"><td><p>minP_EMIC</p></td>
<td><p>p-value of EMIC. When a transcript-leveled EMIC is performed, this is the minimum p-value among all transcripts of the gene.</p></td>
</tr>
<tr class="row-odd"><td><p>Details_EMIC</p></td>
<td><p>Each detailed result has four components in brackets: the number of IVs, the causal effect estimate and its standard error, and the p-value. When a transcript-leveled EMIC is performed, results for each transcript are listed.</p></td>
</tr>
<tr class="row-even"><td><p>Chrom</p></td>
<td><p>Chromosome of the gene</p></td>
</tr>
<tr class="row-odd"><td><p>Pos</p></td>
<td><p>The coordinate of the IV with the lowest GWAS p-value</p></td>
</tr>
<tr class="row-even"><td><p>GWAS_Var_P</p></td>
<td><p>GWAS p-value of the IV</p></td>
</tr>
<tr class="row-odd"><td><p>GWAS_Var_Beta</p></td>
<td><p>The phenotype association effect size of the IV</p></td>
</tr>
<tr class="row-even"><td><p>GWAS_Var_SE</p></td>
<td><p>Standard error of the effect size</p></td>
</tr>
</tbody>
</table>
<p>The columns of <code class="docutils literal notranslate"><span class="pre">t3.emic.gene.var.tsv.gz</span></code> are the same as <code class="docutils literal notranslate"><span class="pre">t3.emic.gene.txt</span></code>. The difference is that, for each gene, in <code class="docutils literal notranslate"><span class="pre">t3.emic.gene.txt</span></code>, only the eQTL with the lowest GWAS p-value is output, while in <code class="docutils literal notranslate"><span class="pre">turorial_3.emic.gene.var.tsv.gz</span></code>, all eQTLs are output. In this tutorial, the file <code class="docutils literal notranslate"><span class="pre">t3.emic.gene.PleiotropyFinemapping.txt</span></code> is empty, we ignore it here.</p>
<p>File <code class="docutils literal notranslate"><span class="pre">t3.qq.png</span></code> saves the Q-Q plot for GWAS p-values of IVs. File <code class="docutils literal notranslate"><span class="pre">t3.emic.qq.png</span></code> saves the Q-Q plot for EMIC p-values.</p>
<p>File <code class="docutils literal notranslate"><span class="pre">t3.scatterplots.emic.pdf</span></code> saves the scatter plots of genetic association with gene expression. Each gene with an EMIC p-value lower than 2.5E-3 (default threshold) is saved on a separate page of the PDF. A filled rectangle on the plots denotes an IV. The red rectangle denotes the most significant GWAS variant among all the IVs of a gene. The slope of the line represents the estimated causal effect. The color of an IV denotes the degree of the LD between the IV and the most significant GWAS variant. The error bars in the rectangles denote the standard errors of the coefficient estimates.</p>
</section>
<section id="gene-based-heritability-estimation">
<span id="t4"></span><h2>4. Gene-based heritability estimation<a class="headerlink" href="#gene-based-heritability-estimation" title="Permalink to this headline"></a></h2>
<p>Heritability is a measure of how well differences in people’s genes account for differences in their phenotypes. This tutorial estimates the heritability of each gene with GWAS summary statistics. The command is:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>java -Xmx4g -jar ../kggsee.jar <span class="se">\</span>
--sum-file scz_gwas_eur_chr1.tsv.gz <span class="se">\</span>
--saved-ref VCFRefhg19 <span class="se">\</span>
--case-col Nca <span class="se">\</span>
--control-col Nco <span class="se">\</span>
--estimate-heritability <span class="se">\</span>
--out t4
</pre></div>
</div>
<p><strong>Explanation of the flags and input files:</strong></p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 80%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--sum-file</span></code></p></td>
<td><p>Specifies a whitespace delimitated file of GWAS summary statistics. In this analysis, in addition to the columns of SNP coordinates and p-values, two columns of case and control sample sizes are also needed.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--saved-ref</span></code></p></td>
<td><p>Specifies the folder of genotypes of reference population in KGGSEE object format, which is saved by the <code class="docutils literal notranslate"><span class="pre">--keep-ref</span></code> flag in the first tutorial.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--case-col</span></code></p></td>
<td><p>Specifies the column name of the case sample size.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--control-col</span></code></p></td>
<td><p>Specifies the column name of the control sample size.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--estimate-heritability</span></code></p></td>
<td><p>Triggers gene-based association tests and estimation of gene heritability.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--out</span></code></p></td>
<td><p>Specifies the prefix of output files.</p></td>
</tr>
</tbody>
</table>
<p><strong>Explanation of the output files:</strong></p>
<p>The output files are generally the same as the first tutorial, except that, in <code class="docutils literal notranslate"><span class="pre">t4.gene.pvalue.txt</span></code>, <code class="docutils literal notranslate"><span class="pre">t4.gene.var.pvalue.txt.gz</span></code>, there are two more columns named Herit and HeritSE, which are the estimate and its standard error of the gene heritability.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="setup.html" class="btn btn-neutral float-left" title="Setup" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="detailed_document.html" class="btn btn-neutral float-right" title="Detailed Document" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019, Miaoxin Li.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>