<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Options &mdash; KGGSEE  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/kggsee.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Detailed Document" href="detailed_document.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> KGGSEE
          </a>
              <div class="version">
                v1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="quick_tutorials.html">Quick tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="detailed_document.html">Detailed Document</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Options</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#reference-population-genotypes">Reference population genotypes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gwas-summary-statistics">GWAS summary statistics</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gene-based-association-and-heritability">Gene-based association and heritability</a></li>
<li class="toctree-l2"><a class="reference internal" href="#dese">DESE</a></li>
<li class="toctree-l2"><a class="reference internal" href="#emic">EMIC</a></li>
<li class="toctree-l2"><a class="reference internal" href="#miscellaneous-global-options">Miscellaneous global options</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">KGGSEE</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Options</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/options.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="options">
<span id="id1"></span><h1>Options<a class="headerlink" href="#options" title="Permalink to this headline"></a></h1>
<p>The options for <a class="reference internal" href="#option-vcf"><span class="std std-ref">Reference population genotypes</span></a>, <a class="reference internal" href="#option-gwas"><span class="std std-ref">GWAS summary statistics</span></a>, and <a class="reference internal" href="#option-misc"><span class="std std-ref">Miscellaneous global options</span></a> act on all analyses. For clarity, we have categorized the other parameters by <a class="reference internal" href="#option-assoc"><span class="std std-ref">Gene-based association and heritability</span></a>, <a class="reference internal" href="#option-dese"><span class="std std-ref">DESE</span></a> and <a class="reference internal" href="#option-emic"><span class="std std-ref">EMIC</span></a>, although this has resulted in some duplication of parameters.</p>
<p>In the “Default” columns of the following tables, “null” denotes that the flag works with an argument but there is no default value; “n/a” denotes that the flag works without an argument.</p>
<section id="reference-population-genotypes">
<span id="option-vcf"></span><h2>Reference population genotypes<a class="headerlink" href="#reference-population-genotypes" title="Permalink to this headline"></a></h2>
<p>These options work on the VCF file of reference population genotypes. Only SNPs that pass the filters will be used for subsequent analyses. These options act on all analyses.</p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 67%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--vcf-ref</span></code></p></td>
<td><p>Specifies a VCF file of genotypes sampled from a reference population. These genotypes are used to estimate LD correlation coefficients among SNPs. For VCF files of separated chromosomes, use wildcards with quotes like <code class="docutils literal notranslate"><span class="pre">&quot;chr*.vcf.gz&quot;</span></code>.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--keep-ref</span></code></p></td>
<td><p>Keep the parsed VCF files as KGGSEE object format in a folder named <code class="docutils literal notranslate"><span class="pre">VCFRefhg*</span></code> under the output folder.</p></td>
<td><p>n/a</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--saved-ref</span></code></p></td>
<td><p>Specifies the folder of genotypes of reference population in KGGSEE object format, which is saved by the <code class="docutils literal notranslate"><span class="pre">--keep-ref</span></code>.  Reading KGGSEE object format files is faster than parsing VCF files.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--filter-maf-le</span></code></p></td>
<td><p>Filter SNPs with a minor allele frequency lower than the setting.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">0.05</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--hwe-all</span></code></p></td>
<td><p>Filter SNPs with a p-value of rejecting Hardy-Weinberg equilibrium lower than the setting.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">1E-5</span></code></p></td>
</tr>
</tbody>
</table>
</section>
<section id="gwas-summary-statistics">
<span id="option-gwas"></span><h2>GWAS summary statistics<a class="headerlink" href="#gwas-summary-statistics" title="Permalink to this headline"></a></h2>
<p>These options work on the GWAS summary statistics and act on all analyses.</p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 67%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--sum-file</span></code></p></td>
<td><p>Specifies a whitespace delimitated file of GWAS summary statistics.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--chrom-col</span></code></p></td>
<td><p>Specifies the column of chromosomes.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--pos-col</span></code></p></td>
<td><p>Specifies the column of coordinates.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">BP</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--p-col</span></code></p></td>
<td><p>Specifies the column of p-values.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">P</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--a1-col</span></code></p></td>
<td><p>Specifies the column of the reference allele to calculate effect sizes.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">A1</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--a2-col</span></code></p></td>
<td><p>Specifies the column of the other allele.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">A2</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--freq-a1-col</span></code></p></td>
<td><p>Specifies the column of the frequency of the allele specified by <code class="docutils literal notranslate"><span class="pre">--a1-col</span></code>.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">FRQ_U</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--beta-col</span></code></p></td>
<td><p>Specifies the column of effect sizes.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--beta-type</span></code></p></td>
<td><p>Specifies the type of effect sizes:  <code class="docutils literal notranslate"><span class="pre">0</span></code> for the linear regression coefficient of a quantitative phenotype; <code class="docutils literal notranslate"><span class="pre">1</span></code> for the logarithm of odds ratio or logistic regression coefficient of a qualitative phenotype; <code class="docutils literal notranslate"><span class="pre">2</span></code> for an odds ratio of a qualitative phenotype.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--se-col</span></code></p></td>
<td><p>Specifies the column of standard errors of effect sizes. Note: even if the effect size is provided as an odds ratio, this is still the standard error of the logarithm (base e) of the odds ratio.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">SE</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--nmiss-col</span></code></p></td>
<td><p>Specifies the column of sample sizes for a quantitative phenotype.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">Neff</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--case-col</span></code></p></td>
<td><p>Specifies the column of case sample sizes for a qualitative phenotype.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--control-col</span></code></p></td>
<td><p>Specifies the column of control sample sizes for a qualitative phenotype.</p></td>
<td><p>null</p></td>
</tr>
</tbody>
</table>
</section>
<section id="gene-based-association-and-heritability">
<span id="option-assoc"></span><h2>Gene-based association and heritability<a class="headerlink" href="#gene-based-association-and-heritability" title="Permalink to this headline"></a></h2>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 67%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--gene-assoc</span></code></p></td>
<td><p>Triggers gene-based association tests.</p></td>
<td><p>n/a</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--estimate-heritability</span></code></p></td>
<td><p>Triggers gene-based association tests and estimation of gene heritability.</p></td>
<td><p>n/a</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--prevalence</span></code></p></td>
<td><p>Specifies the proportion of cases in the population when estimating the heritability of a qualitative phenotype.</p></td>
<td><p>0.01</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--neargene</span></code></p></td>
<td><p>Specifies the number of basepairs to extend at both ends of a gene, when considering SNPs belonging to the gene.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">5000</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--eqtl-file</span></code></p></td>
<td><p>Specifies a fasta-styled file of eQTL summary statistics. If this flag is used, <code class="docutils literal notranslate"><span class="pre">--neargene</span></code> is overridden, and eQTLs of a gene or transcript will be grouped and tested.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--filter-eqtl-p</span></code></p></td>
<td><p>Specifies the threshold of eQTL p-values. Only eQTLs with a p-value lower than the threshold will be used. The default is <code class="docutils literal notranslate"><span class="pre">0.01</span></code> when performing gene-based association tests and heritability estimating.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">0.01</span></code></p></td>
</tr>
</tbody>
</table>
</section>
<section id="dese">
<span id="option-dese"></span><h2>DESE<a class="headerlink" href="#dese" title="Permalink to this headline"></a></h2>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 67%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--gene-finemapping</span></code></p></td>
<td><p>Trigers the DESE, eDESE or SelDP.</p></td>
<td><p>n/a</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--expression-file</span></code></p></td>
<td><p>Specifies a gene expression file that contains means and standard errors of gene expressions in multiple tissues.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--multiple-testing</span></code></p></td>
<td><p>Specifies the method for multiple testing correction. <code class="docutils literal notranslate"><span class="pre">bonf</span></code> denotes performing Bonferroni correction; <code class="docutils literal notranslate"><span class="pre">benfdr</span></code> denotes controlling false discovery rate by the Benjamini–Hochberg method; <code class="docutils literal notranslate"><span class="pre">fixed</span></code> denotes no correction.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">bonf</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--p-value-cutoff</span></code></p></td>
<td><p>Specifies the threshold of the adjusted p-value for fine-mapping. Only genes with an adjusted p-value lower than the threshold will be retained for fine-mapping.</p></td>
<td><p>0.05</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--top-gene</span></code></p></td>
<td><p>Specifies the maximum number of genes with the smallest p-values that will be retained for fine-mapping.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--geneset-db</span></code></p></td>
<td><p>Specifies <a class="reference external" href="http://www.gsea-msigdb.org/gsea/msigdb/index.jsp">MSigDB</a> gene sets for enrichment analysis:</p>
<p><code class="docutils literal notranslate"><span class="pre">cura</span></code>: C2. curated gene sets;</p>
<p><code class="docutils literal notranslate"><span class="pre">cgp</span></code>: C2. chemical and genetic perturbations;</p>
<p><code class="docutils literal notranslate"><span class="pre">cano</span></code>: C2. canonical pathways;</p>
<p><code class="docutils literal notranslate"><span class="pre">cmop</span></code>: C4. computational gene sets;</p>
<p><code class="docutils literal notranslate"><span class="pre">onto</span></code>: C5. ontology gene sets;</p>
<p><code class="docutils literal notranslate"><span class="pre">onco</span></code>: C6. oncogenic signature gene sets;</p>
<p><code class="docutils literal notranslate"><span class="pre">immu</span></code>: C7. immunologic signature gene sets.</p>
</td>
<td><p>null</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--geneset-file</span></code></p></td>
<td><p>Specifies a user-defined file of gene sets for enrichment analysis.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--neargene</span></code></p></td>
<td><p>Specifies the number of basepairs to extend at both ends of a gene, when considering SNPs belonging to the gene.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">5000</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--eqtl-file</span></code></p></td>
<td><p>Specifies a fasta-styled file of eQTL summary statistics. If this flag is used, <code class="docutils literal notranslate"><span class="pre">--neargene</span></code> is overridden, and eQTLs of a gene or transcript will be grouped and tested.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--filter-eqtl-p</span></code></p></td>
<td><p>Specifies the threshold of eQTL p-values. Only eQTLs with a p-value lower than the threshold will be used. The default is <code class="docutils literal notranslate"><span class="pre">0.01</span></code> when performing DESE.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">0.01</span></code></p></td>
</tr>
</tbody>
</table>
</section>
<section id="emic">
<span id="option-emic"></span><h2>EMIC<a class="headerlink" href="#emic" title="Permalink to this headline"></a></h2>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 67%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--emic</span></code></p></td>
<td><p>Triggers the EMIC.</p></td>
<td><p>n/a</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--eqtl-file</span></code></p></td>
<td><p>Specifies a fasta-styled file of eQTL summary statistics.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--filter-eqtl-p</span></code></p></td>
<td><p>Specifies the threshold of eQTL p-values. Only eQTLs with a p-value lower than the threshold will be used. The default is <code class="docutils literal notranslate"><span class="pre">1E-4</span></code> when performing EMIC.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">1E-4</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--ld-pruning-mr</span></code></p></td>
<td><p>Specifies the threshold of LD coefficient when pruning variants. For each gene or transcript, eQTLs with LD coefficients higher than the threshold will be pruned.</p></td>
<td><p>0.5</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--emic-pfm-p</span></code></p></td>
<td><p>Specifies the p-value threshold to further perform an EMIC pleiotropy fine-mapping (EMIC-PFM) analysis. If the EMIC p-value of a gene is lower than the threshold, an EMIC-PFM will be performed to control the false-positive caused by pleiotropy.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">2.5E-6</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--emic-plot-p</span></code></p></td>
<td><p>Specifies the p-value threshold for plotting a scatter plot. Genes with an EMIC p-value lower than the threshold will be plotted.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">2.5E-3</span></code></p></td>
</tr>
</tbody>
</table>
</section>
<section id="miscellaneous-global-options">
<span id="option-misc"></span><h2>Miscellaneous global options<a class="headerlink" href="#miscellaneous-global-options" title="Permalink to this headline"></a></h2>
<p>These options act on all analyses.</p>
<table class="colwidths-given tight-table docutils align-default">
<colgroup>
<col style="width: 25%" />
<col style="width: 67%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Flag</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--nt</span></code></p></td>
<td><p>Specifies the number of threads.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">4</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--buildver</span></code></p></td>
<td><p>Specifies the reference genome version of the coordinates. The supported versions are <code class="docutils literal notranslate"><span class="pre">hg19</span></code> and <code class="docutils literal notranslate"><span class="pre">hg38</span></code>.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">hg19</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--db-gene</span></code></p></td>
<td><p>Specifies the database of gene annotations. <code class="docutils literal notranslate"><span class="pre">refgene</span></code> for RefSeq Genes; <code class="docutils literal notranslate"><span class="pre">gencode</span></code> for GENCODE; <code class="docutils literal notranslate"><span class="pre">refgene,gencode</span></code> for both.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">gencode</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--excel</span></code></p></td>
<td><p>Output results in Excel format.</p></td>
<td><p>n/a</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--only-hgnc-gene</span></code></p></td>
<td><p>Only genes with an HGNC-approved gene symbol are considered in analyses.</p></td>
<td><p>n/a</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--out</span></code></p></td>
<td><p>Specifies the output prefix of results.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--regions-out</span></code></p></td>
<td><p>Specifies genomic regions to be excluded in analyses, e.g. <code class="docutils literal notranslate"><span class="pre">chr1,chr2:2323-34434,chr2:43455-345555</span></code>.</p></td>
<td><p>null</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--resource</span></code></p></td>
<td><p>Specifies the path KGGSEE running resource data.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">path/to/kggsee.jar/resources/</span></code></p></td>
</tr>
</tbody>
</table>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="detailed_document.html" class="btn btn-neutral float-left" title="Detailed Document" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019, Miaoxin Li.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>