%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english,openany,oneside]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{3}
\setcounter{secnumdepth}{3}



\title{KGGSEE V1.1 User Manual}
\date{Apr 15, 2022}
\release{}
\author{Miaoxin Li, Lin Jiang, Xiangyi Li, Lin Miao}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\sphinxstepscope


\chapter{Setup}
\label{\detokenize{setup:setup}}\label{\detokenize{setup:id1}}\label{\detokenize{setup::doc}}

\section{System requirements}
\label{\detokenize{setup:system-requirements}}

\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{3}{10}|\X{7}{10}|}
\hline

\sphinxAtStartPar
Operating system
&
\sphinxAtStartPar
KGGSEE runs in a Java Virtual Machine. It does not matter which operating system it runs in.
\\
\hline
\sphinxAtStartPar
Java Runtime Environment
&
\sphinxAtStartPar
A Java SE Runtime Environment of version 1.8 or higher is needed.
\\
\hline
\sphinxAtStartPar
CPU
&
\sphinxAtStartPar
A CPU with four cores or more is recommended.
\\
\hline
\sphinxAtStartPar
Memory
&
\sphinxAtStartPar
16 GB RAM or higher is recommended.
\\
\hline
\sphinxAtStartPar
Free space
&
\sphinxAtStartPar
KGGSEE and related datasets may take up to 10 GB.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\section{Setup the Java Runtime Environment (JRE)}
\label{\detokenize{setup:setup-the-java-runtime-environment-jre}}
\sphinxAtStartPar
KGGSEE needs JRE 1.8 or higher to run. Both \sphinxhref{https://java.com/en/download/manual.jsp}{Java(TM) SE JRE} and \sphinxhref{https://openjdk.java.net/install}{OpenJDK JRE} are competent for KGGSEE. Please follow the instructions on the websites to complete the installation and also add Java to the system PATH.

\sphinxAtStartPar
Check the JRE by entering \sphinxcode{\sphinxupquote{java \sphinxhyphen{}version}} in a Terminal of Linux or MacOS, or CMD or PowerShell of Windows. If it displays the JRE version like \sphinxcode{\sphinxupquote{Java(TM) SE Runtime Environment (build 1.8.0\_xxx)}} or \sphinxcode{\sphinxupquote{OpenJDK Runtime Environment (build 1.8.0\_xxx)}}, it means the JRE has already been set up. Otherwise, check if JRE has been installed and if Java is in the system PATH.


\newpage
\section{Setup KGGSEE}
\label{\detokenize{setup:setup-kggsee}}
\sphinxAtStartPar
Download the bundled file of \sphinxcode{\sphinxupquote{kggsee.jar}}, running resource dataset and quick tutorial dataset from \sphinxhref{http://pmglab.top/kggsee/\#/download}{the download page} and unzip.

\sphinxAtStartPar
The running resource dataset includes:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{7}{13}|\X{6}{13}|}
\hline

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/hg19/kggseqv1.1\_hg19\_GEncode.txt.gz}}
&
\sphinxAtStartPar
hg19 \sphinxhref{https://www.gencodegenes.org}{GENCODE} annotation
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/hg19/kggseqv1.1\_hg19\_refGene.txt.gz}}
&
\sphinxAtStartPar
hg19 \sphinxhref{https://www.ncbi.nlm.nih.gov/refseq/rsg}{RefGene} annotation
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/hg38/kggseqv1.1\_hg38\_GEncode.txt.gz}}
&
\sphinxAtStartPar
hg38 \sphinxhref{https://www.gencodegenes.org}{GENCODE} annotation
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/hg38/kggseqv1.1\_hg38\_refGene.txt.gz}}
&
\sphinxAtStartPar
hg38 \sphinxhref{https://www.ncbi.nlm.nih.gov/refseq/rsg}{RefGene} annotation
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/HgncGene.txt.gz}}
&
\sphinxAtStartPar
\sphinxhref{https://www.genenames.org}{HGNC} gene ID
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/ENSTGene.gz}}
&
\sphinxAtStartPar
\sphinxhref{https://www.ensembl.org/index.html}{Ensembl} gene ID and transcript ID
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{GTEx_v8_TMM.gene.meanSE.txt.gz}}
&
\sphinxAtStartPar
The gene\sphinxhyphen{}level expression profile of the \sphinxhref{https://www.gtexportal.org/home/}{GTEx v8} tissues
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{GTEx_v8_TMM.transcript.meanSE.txt}}
&
\sphinxAtStartPar
The transcript\sphinxhyphen{}level expression profile of the \sphinxhref{https://www.gtexportal.org/home/}{GTEx v8} tissues
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/HCL\_scRNA\_cluster\_mean.tsv.gz}}
&
\sphinxAtStartPar
The expression profile of cell clusters generated from the scRNA\sphinxhyphen{}seq dataset of the \sphinxhref{bis.zju.edu.cn/HCL/}{Human Cell Landscape}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/*.symbols.gmt.gz}}
&
\sphinxAtStartPar
\sphinxhref{http://www.gsea-msigdb.org/gsea/msigdb/index.jsp}{MSigDB} gene sets
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
The tutorial dataset includes:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{7}{13}|\X{6}{13}|}
\hline

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{tutorials/scz\_gwas\_eur\_chr1.tsv.gz}}
&
\sphinxAtStartPar
Summary statistics of chr1 SNPs for a GWAS of schizophrenia on the European population
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{tutorials/1kg\_hg19\_eur\_chr1.vcf.gz}}
&
\sphinxAtStartPar
Genotypes of chr1 SNPs sampled from the 1000 Genome Project European population
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{tutorials/GTEx\_v8\_gene\_BrainBA9.eqtl.txt.gz}}
&
\sphinxAtStartPar
Summary statistics of eQTLs calculated from gene\sphinxhyphen{}level expression profile of GTEx v8 brain BA9
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{tutorials/GTEx\_v8\_transcript\_BrainBA9.eqtl.txt.gz}}
&
\sphinxAtStartPar
Summary statistics of eQTLs calculated from transcript\sphinxhyphen{}level expression profile of GTEx v8 brain BA9
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
For running customized analyses, the following data is needed, refer to {\hyperref[\detokenize{detailed_document:detailed-document}]{\sphinxcrossref{\DUrole{std,std-ref}{Detailed Document}}}} for descriptions of file formats.
\begin{itemize}
\item {} 
\sphinxAtStartPar
A file of GWAS summary statistics of the phenotype to be studied.

\item {} 
\sphinxAtStartPar
VCF files of genotypes sampled from the population of the GWAS to be studied. Genotypes of the 1000 Genomes Project Phase3 v5 can be downloaded from the \sphinxhref{ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20130502}{NCBI FTP site} or the \sphinxhref{ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502}{1000 Genomes Project FTP site}.

\item {} 
\sphinxAtStartPar
A file of eQTL summary statistics calculated from target tissues may be used. We provide gene\sphinxhyphen{}based and transcript\sphinxhyphen{}based eQTL summary statistics for GTEx v8 tissues available for download (refer to \sphinxhref{http://pmglab.top/kggsee/\#/download}{the download page}).

\end{itemize}

\sphinxstepscope


\chapter{Quick tutorials}
\label{\detokenize{quick_tutorials:quick-tutorials}}\label{\detokenize{quick_tutorials:id1}}\label{\detokenize{quick_tutorials::doc}}
\sphinxAtStartPar
We provide four quick tutorials; each shows one function of KGGSEE. In each tutorial, we provide the command line and a brief explanation of flags and output files. Please refer to {\hyperref[\detokenize{detailed_document:detailed-document}]{\sphinxcrossref{\DUrole{std,std-ref}{Detailed Document}}}} and {\hyperref[\detokenize{options:options}]{\sphinxcrossref{\DUrole{std,std-ref}{Options}}}} for details. The first tutorial ({\hyperref[\detokenize{quick_tutorials:t1}]{\sphinxcrossref{\DUrole{std,std-ref}{Gene\sphinxhyphen{}based association tests}}}}) should be done first, then there is no need to do the following in order.

\sphinxAtStartPar
Make sure the KGGSEE Java archive \sphinxcode{\sphinxupquote{kggsee.jar}}, the running resource data folder \sphinxcode{\sphinxupquote{resources/}}, and the tutorial data folder \sphinxcode{\sphinxupquote{tutorials/}} are under the same folder. For convenience, enter the \sphinxcode{\sphinxupquote{tutorials/}} directory first.


\section{Gene\sphinxhyphen{}based association tests}
\label{\detokenize{quick_tutorials:gene-based-association-tests}}\label{\detokenize{quick_tutorials:t1}}
\sphinxAtStartPar
GATES and ECS are two statistical methods combining the p\sphinxhyphen{}values of a group of SNPs into one p\sphinxhyphen{}value. This analysis inputs p\sphinxhyphen{}values of SNPs and outputs p\sphinxhyphen{}values of genes. The command is:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}keep\PYGZhy{}ref \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}assoc \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t1
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxstylestrong{Explanation of the flags and input files:}


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a whitespace delimitated file of GWAS summary statistics. In this analysis, columns of SNP coordinates and p\sphinxhyphen{}values (CHR, BP, and P by default) are needed.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}vcf\sphinxhyphen{}ref}}
&
\sphinxAtStartPar
Specifies a VCF file of genotypes sampled from a reference population. These genotypes are used to estimate LD correlation coefficients among SNPs.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}keep\sphinxhyphen{}ref}}
&
\sphinxAtStartPar
Keep the parsed VCF file (KGGSEE object format) in a folder named \sphinxcode{\sphinxupquote{VCFRefhg19}} under the output folder. KGGSEE will read these files in the following tutorials, which will be faster than parsing VCF files.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}assoc}}
&
\sphinxAtStartPar
Triggers gene\sphinxhyphen{}based association tests.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}out}}
&
\sphinxAtStartPar
Specifies the prefix of output files.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\newpage
\sphinxAtStartPar
\sphinxstylestrong{Explanation of the output files:}

\sphinxAtStartPar
The numeric results of gene\sphinxhyphen{}based association tests are saved in \sphinxcode{\sphinxupquote{t1.gene.pvalue.txt}}. There are seven columns in the file:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Gene
&
\sphinxAtStartPar
Gene symbol
\\
\hline
\sphinxAtStartPar
\#Var
&
\sphinxAtStartPar
Number of variants within the gene
\\
\hline
\sphinxAtStartPar
ECSP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of ECS
\\
\hline
\sphinxAtStartPar
GATESP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of GATES
\\
\hline
\sphinxAtStartPar
Chrom
&
\sphinxAtStartPar
Chromosome of the gene
\\
\hline
\sphinxAtStartPar
Pos
&
\sphinxAtStartPar
Coordinate of the variant with the lowest p\sphinxhyphen{}value within the gene
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_P
&
\sphinxAtStartPar
p\sphinxhyphen{}value of the variant
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
The columns of \sphinxcode{\sphinxupquote{t1.gene.var.pvalue.txt.gz}} are the same as \sphinxcode{\sphinxupquote{t1.gene.pvalue.txt}}. The difference is that, for each gene, in \sphinxcode{\sphinxupquote{t1.gene.pvalue.txt}}, only the variant with the lowest p\sphinxhyphen{}value is output, while in \sphinxcode{\sphinxupquote{t1.gene.var.pvalue.txt.gz}}, all variants are output.

\sphinxAtStartPar
The Q\sphinxhyphen{}Q plots for p\sphinxhyphen{}values of inputted GWAS file (inside or outside of gene) and gene\sphinxhyphen{}based association tests by GATES or ECS are saved in \sphinxcode{\sphinxupquote{t1.qq.png}}.


\section{DESE}
\label{\detokenize{quick_tutorials:dese}}\label{\detokenize{quick_tutorials:t2}}
\sphinxAtStartPar
DESE performs phenotype\sphinxhyphen{}tissue association tests and conditional gene\sphinxhyphen{}based association tests at the same time. This analysis inputs p\sphinxhyphen{}values of a GWAS and expression profile of multiple tissues; outputs p\sphinxhyphen{}values of phenotype\sphinxhyphen{}tissue associations and conditional p\sphinxhyphen{}values of genes. The command is:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}saved\PYGZhy{}ref VCFRefhg19 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}expression\PYGZhy{}file GTEx_v8_TMM.gene.meanSE.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}finemapping \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t2
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxstylestrong{Explanation of the flags and input files:}


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{4}|\X{3}{4}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a whitespace delimitated file of GWAS summary statistics. In this analysis, columns of SNP coordinates and p\sphinxhyphen{}values are needed.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}saved\sphinxhyphen{}ref}}
&
\sphinxAtStartPar
Specifies the folder of genotypes of reference population in KGGSEE object format, which is saved by the \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}keep\sphinxhyphen{}ref}} flag in the first tutorial.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}expression\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a gene expression file that contains means and standard errors of gene expressions in multiple tissues/cell types. Here \sphinxcode{\sphinxupquote{GTEx_v8_TMM.gene.meanSE.txt.gz}} is for gene\sphinxhyphen{}level DESE. Try \sphinxcode{\sphinxupquote{GTEx_v8_TMM.transcript.meanSE.txt}} for transcript\sphinxhyphen{}level DESE; try \sphinxcode{\sphinxupquote{HCL\_scRNA\_cluster\_mean.tsv.gz}} for cell\sphinxhyphen{}cluster based DESE.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}finemapping}}
&
\sphinxAtStartPar
Triggers the DESE analysis.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}out}}
&
\sphinxAtStartPar
Specifies the prefix of output files.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\newpage
\sphinxAtStartPar
\sphinxstylestrong{Explanation of the output files:}

\sphinxAtStartPar
The three files of \sphinxcode{\sphinxupquote{t2.gene.pvalue.txt}}, \sphinxcode{\sphinxupquote{t2.gene.var.pvalue.txt.gz}}, and \sphinxcode{\sphinxupquote{t2.qq.png}} are the same as their counterparts with the same suffixes of the first tutorial. In addition, the results of conditional gene\sphinxhyphen{}based association tests are in \sphinxcode{\sphinxupquote{t2.finemapping.gene.ecs.txt}} which contains nine columns:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Gene
&
\sphinxAtStartPar
Gene symbol
\\
\hline
\sphinxAtStartPar
Chrom
&
\sphinxAtStartPar
Chromosome of the gene
\\
\hline
\sphinxAtStartPar
StartPos
&
\sphinxAtStartPar
Start coordinate of the gene
\\
\hline
\sphinxAtStartPar
EndPos
&
\sphinxAtStartPar
End coordinate of the gene
\\
\hline
\sphinxAtStartPar
\#Var
&
\sphinxAtStartPar
Number of variants within the gene
\\
\hline
\sphinxAtStartPar
Group
&
\sphinxAtStartPar
LD group number. Conditional ECS tests were performed for genes within a same LD group.
\\
\hline
\sphinxAtStartPar
ECSP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of ECS
\\
\hline
\sphinxAtStartPar
CondiECSP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of conditional gene\sphinxhyphen{}based association tests by conditional ECS
\\
\hline
\sphinxAtStartPar
GeneScore
&
\sphinxAtStartPar
The gene’s selective expression score in all tissues. A gene with a high socre will be given higher priority to enter the conditioning procedure.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
Results of phenotype\sphinxhyphen{}tissue associations are in \sphinxcode{\sphinxupquote{t2.celltype.txt}}. This is basically a Wilcoxon rank\sphinxhyphen{}sum test which tests whether the selective expression median of the phenotype\sphinxhyphen{}associated genes is significantly higher than that of other genes in an interrogated tissue. The file contains three columns:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
TissueName
&
\sphinxAtStartPar
Name of the tissue being tested
\\
\hline
\sphinxAtStartPar
PValue
&
\sphinxAtStartPar
p\sphinxhyphen{}values of phenotype\sphinxhyphen{}tissue associations.
\\
\hline
\sphinxAtStartPar
Log(p)
&
\sphinxAtStartPar
The negative logarithm (base 10) of p\sphinxhyphen{}values of phenotype\sphinxhyphen{}tissue associations
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\section{EMIC}
\label{\detokenize{quick_tutorials:emic}}\label{\detokenize{quick_tutorials:t3}}
\sphinxAtStartPar
EMIC inferences gene expressions’ causal effect on a complex phenotype with dependent expression quantitative loci by a robust median\sphinxhyphen{}based Mendelian randomization. SNPs with effects on both the phenotype and a gene are considered instrumental variables (IVs) of the gene, which can be used to infer the gene’s expression effect on the phenotype. This analysis inputs effect sizes of SNPs on the phenotype and genes’ expressions; outputs effect sizes and p\sphinxhyphen{}values of genes’ expression effects on the phenotype. The command is:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}saved\PYGZhy{}ref VCFRefhg19 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}gene\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}beta\PYGZhy{}col OR \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}beta\PYGZhy{}type \PYG{l+m}{2} \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}emic \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t3
\end{sphinxVerbatim}

\newpage
\sphinxAtStartPar
\sphinxstylestrong{Explanation of the flags and input files:}


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Flag
&
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a whitespace delimitated file of GWAS summary statistics. In this analysis, in addition to the columns of SNP coordinates and p\sphinxhyphen{}values, two columns of SNP alleles (named A1 and A2 by default), a column of A1 allele frequency (named FRQ\_U by default), and two columns of SNP effect sizes (no default header) and their standard errors (named SE by default) are also needed.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}saved\sphinxhyphen{}ref}}
&
\sphinxAtStartPar
Specifies the folder of genotypes of reference population in KGGSEE object format, which is saved by the \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}keep\sphinxhyphen{}ref}} flag in the first tutorial.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a fasta\sphinxhyphen{}styled file of SNPs’ effects on gene expressions.

Here \sphinxcode{\sphinxupquote{GTEx\_v8\_gene\_BrainBA9.eqtl.txt.gz}} for gene\sphinxhyphen{}level EMIC.

Try \sphinxcode{\sphinxupquote{GTEx\_v8\_transcript\_BrainBA9.eqtl.txt.gz}} for transcript\sphinxhyphen{}level EMIC.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}beta\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column name of effect sizes in the GWAS file.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}beta\sphinxhyphen{}type}}
&
\sphinxAtStartPar
Specifies the type of the effect size; here \sphinxcode{\sphinxupquote{2}} means that it is the odds ratio for a qualitative phenotype.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic}}
&
\sphinxAtStartPar
Triggers the EMIC analysis.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}out}}
&
\sphinxAtStartPar
Specifies the prefix of output files.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
\sphinxstylestrong{Explanation of the output files:}

\sphinxAtStartPar
The numeric results of EMIC are saved in \sphinxcode{\sphinxupquote{t3.emic.gene.txt}}. There are nine columns in the file:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Gene
&
\sphinxAtStartPar
The gene symbol
\\
\hline
\sphinxAtStartPar
\#Var
&
\sphinxAtStartPar
Number of IVs within the gene
\\
\hline
\sphinxAtStartPar
minP\_EMIC
&
\sphinxAtStartPar
p\sphinxhyphen{}value of EMIC. When a transcript\sphinxhyphen{}level EMIC is performed, this is the minimum p\sphinxhyphen{}value among all transcripts of the gene.
\\
\hline
\sphinxAtStartPar
Details\_EMIC
&
\sphinxAtStartPar
Each detailed result has four components in brackets: the number of IVs, the causal effect estimate and its standard error, and the p\sphinxhyphen{}value. When a transcript\sphinxhyphen{}level EMIC is performed, results for each transcript are listed.
\\
\hline
\sphinxAtStartPar
Chrom
&
\sphinxAtStartPar
Chromosome of the gene
\\
\hline
\sphinxAtStartPar
Pos
&
\sphinxAtStartPar
The coordinate of the IV with the lowest GWAS p\sphinxhyphen{}value
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_P
&
\sphinxAtStartPar
GWAS p\sphinxhyphen{}value of the IV
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_Beta
&
\sphinxAtStartPar
The phenotype association effect size of the IV
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_SE
&
\sphinxAtStartPar
Standard error of the effect size
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
The columns of \sphinxcode{\sphinxupquote{t3.emic.gene.var.tsv.gz}} are the same as \sphinxcode{\sphinxupquote{t3.emic.gene.txt}}. The difference is that, for each gene, in \sphinxcode{\sphinxupquote{t3.emic.gene.txt}}, only the eQTL with the lowest GWAS p\sphinxhyphen{}value is output, while in \sphinxcode{\sphinxupquote{turorial\_3.emic.gene.var.tsv.gz}}, all eQTLs are output. In this tutorial, the file \sphinxcode{\sphinxupquote{t3.emic.gene.PleiotropyFinemapping.txt}} is empty, we ignore it here.

\sphinxAtStartPar
File \sphinxcode{\sphinxupquote{t3.qq.png}} saves the Q\sphinxhyphen{}Q plot for GWAS p\sphinxhyphen{}values of IVs. File \sphinxcode{\sphinxupquote{t3.emic.qq.png}} saves the Q\sphinxhyphen{}Q plot for EMIC p\sphinxhyphen{}values. File \sphinxcode{\sphinxupquote{t3.scatterplots.emic.pdf}} saves the scatter plots of genetic association with gene expression. Each gene with an EMIC p\sphinxhyphen{}value lower than 2.5E\sphinxhyphen{}3 (default threshold) is saved on a separate page of the PDF. A filled rectangle on the plots denotes an IV. The red rectangle denotes the most significant GWAS variant among all the IVs of a gene. The slope of the line represents the estimated causal effect. The color of an IV denotes the degree of the LD between the IV and the most significant GWAS variant. The error bars in the rectangles denote the standard errors of the coefficient estimates.


\section{Gene\sphinxhyphen{}based heritability estimation}
\label{\detokenize{quick_tutorials:gene-based-heritability-estimation}}\label{\detokenize{quick_tutorials:t4}}
\sphinxAtStartPar
Heritability is a measure of how well differences in people’s genes account for differences in their phenotypes. This tutorial estimates the heritability of each gene with GWAS summary statistics. The command is:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}saved\PYGZhy{}ref VCFRefhg19 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}case\PYGZhy{}col Nca \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}control\PYGZhy{}col Nco \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}estimate\PYGZhy{}heritability \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t4
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxstylestrong{Explanation of the flags and input files:}


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{2}{7}|\X{5}{7}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a whitespace delimitated file of GWAS summary statistics. In this analysis, in addition to the columns of SNP coordinates and p\sphinxhyphen{}values, two columns of case and control sample sizes are also needed.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}saved\sphinxhyphen{}ref}}
&
\sphinxAtStartPar
Specifies the folder of genotypes of reference population in KGGSEE object format, which is saved by the \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}keep\sphinxhyphen{}ref}} flag in the first tutorial.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}case\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column name of the case sample size.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}control\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column name of the control sample size.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}estimate\sphinxhyphen{}heritability}}
&
\sphinxAtStartPar
Triggers gene\sphinxhyphen{}based association tests and estimation of gene heritability.
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}out}}
&
\sphinxAtStartPar
Specifies the prefix of output files.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
\sphinxstylestrong{Explanation of the output files:}

\sphinxAtStartPar
The output files are generally the same as the first tutorial, except that, in \sphinxcode{\sphinxupquote{t4.gene.pvalue.txt}}, \sphinxcode{\sphinxupquote{t4.gene.var.pvalue.txt.gz}}, there are two more columns named Herit and HeritSE, which are the estimate and its standard error of the gene heritability.

\sphinxstepscope


\chapter{Detailed Document}
\label{\detokenize{detailed_document:detailed-document}}\label{\detokenize{detailed_document:id1}}\label{\detokenize{detailed_document::doc}}
\sphinxAtStartPar
We first describe the general aspects of all analyses and then describe details for each analysis. KGGSEE performs analysis according to the following procedure:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Reads genotypes of an ancestrally matched reference population, e.g., a panel of 1000 Genomes Project. The genotypes can be in a VCF file specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}vcf\sphinxhyphen{}ref}}, and if \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}keep\sphinxhyphen{}ref}} is used at the same run, KGGSEE  saves the parsed VCF file in KGGSEE object format in the folder of \sphinxcode{\sphinxupquote{path/to/outputs/VCFRefhg*/}}. For later run with \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}keep\sphinxhyphen{}ref path/to/outputs/VCFRefhg*/}}, KGGSEE reads genotypes from the object format files, which will be faster than parsing VCF files. KGGSEE calculates the minor allele frequency of each SNP and filters out SNPs with a minor allele frequency lower than the threshold specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}maf\sphinxhyphen{}le}} (default: \sphinxcode{\sphinxupquote{0.05}}). KGGSEE also calculates the p\sphinxhyphen{}value of rejecting Hardy\sphinxhyphen{}Weinberg equilibrium for each SNP and filters out SNPs with a p\sphinxhyphen{}value lower than the threshold specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}hwe\sphinxhyphen{}all}} (default: \sphinxcode{\sphinxupquote{1E\sphinxhyphen{}5}}). Only SNPs with genotypes of the reference population and who have passed the two filters will be considered in the following procedures.

\item {} 
\sphinxAtStartPar
Reads GWAS summary statistics from a whitespace delimitated file specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}}. Depending on the analysis performed, this file needs to have different columns, which we will describe separately below. For all analyses, an eQTL summary statistic file specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}} may be read. We provide gene\sphinxhyphen{}based and transcript\sphinxhyphen{}based eQTL summary statistics for GTEx v8 tissues available for downloading on \sphinxhref{https://mailsysueducn-my.sharepoint.com/personal/limiaoxin\_mail\_sysu\_edu\_cn/\_layouts/15/onedrive.aspx?id=\%2Fpersonal\%2Flimiaoxin\%5Fmail\%5Fsysu\%5Fedu\%5Fcn\%2FDocuments\%2Ftools\%2Fkggsee\%2Fresources\&ga=1}{OneDrive}.

\item {} 
\sphinxAtStartPar
Based on the flag specified, KGGSEE reads more needed files and performs the corresponding analysis.
\begin{itemize}
\item {} 
\sphinxAtStartPar
{\hyperref[\detokenize{detailed_document:detail-ecs}]{\sphinxcrossref{\DUrole{std,std-ref}{Gene\sphinxhyphen{}based association tests}}}} trigered by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}assoc}};

\item {} 
\sphinxAtStartPar
{\hyperref[\detokenize{detailed_document:detail-dese}]{\sphinxcrossref{\DUrole{std,std-ref}{DESE}}}} trigered by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}finemapping}};

\item {} 
\sphinxAtStartPar
{\hyperref[\detokenize{detailed_document:detail-emic}]{\sphinxcrossref{\DUrole{std,std-ref}{EMIC}}}} trigered by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic}};

\item {} 
\sphinxAtStartPar
{\hyperref[\detokenize{detailed_document:detail-h2}]{\sphinxcrossref{\DUrole{std,std-ref}{Gene\sphinxhyphen{}based heritability estimation}}}} trigered by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}estimate\sphinxhyphen{}heritability}}.

\end{itemize}

\end{enumerate}
\phantomsection\label{\detokenize{detailed_document:eqtl-file}}
\sphinxAtStartPar
The KGGSEE format of eQTL summary statistics is fasta\sphinxhyphen{}styled. An example is as follows:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
#symbol   id    chr     pos     ref     alt     altfreq beta    se     p      neff   r2
>WASH7P ENSG00000227232 1
52238   T       G       0.94    -1.77   0.28    5.1E-9  65      0.38
74681   G       T       0.95    -1.45   0.33    1.1E-5  63      0.23
92638   A       T       0.24    0.54    0.20    7.9E-3  53      0.12
>MIR130 ENSG00000284557 1
52238   T       G       0.94    -1.77   0.28    5.1E-9  65      0.38
74681   G       T       0.95    -1.45   0.33    1.1E-5  63      0.23
\end{sphinxVerbatim}

\sphinxAtStartPar
The first row starting with \sphinxcode{\sphinxupquote{\#}} is the header line. Then, eQTLs of each gene/transcript are chunked. For each gene/transcript, the first row has three columns of (1) the gene symbol prefixed by \sphinxcode{\sphinxupquote{\textgreater{}}}, (2) Ensembl gene/transcript ID, and (3) chromosome; the second and following rows have nine columns of (4) the eQTL coordinate, (5) the reference allele, (6) the alternative allele, (7) the frequency of the alternative allele, (8) the effect size, (9) the standard error of the effect size, (10) the p\sphinxhyphen{}value of nonzero effect size, (11) the effective sample size and (12) coefficient of determination.


\section{Gene\sphinxhyphen{}based association tests}
\label{\detokenize{detailed_document:gene-based-association-tests}}\label{\detokenize{detailed_document:detail-ecs}}
\sphinxAtStartPar
KGGSEE performs the gene\sphinxhyphen{}based association analysis by GATES (a rapid and powerful \sphinxstylestrong{G}ene\sphinxhyphen{}based \sphinxstylestrong{A}ssociation \sphinxstylestrong{T}est using \sphinxstylestrong{E}xtended \sphinxstylestrong{S}imes procedure) and ECS (an \sphinxstylestrong{E}ffective \sphinxstylestrong{C}hi\sphinxhyphen{}square \sphinxstylestrong{S} tatistics). The \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}assoc}} flag trigers both.

\sphinxAtStartPar
GATES (\sphinxhref{https://doi.org/10.1016/j.ajhg.2011.01.019}{the GATES paper}) is basically an extension of the Simes procedure to dependent tests, as the individual GWAS tests are dependent due to LD. GATES calculates an effective number of independent p\sphinxhyphen{}values which is then used by a Simes procedure.

\sphinxAtStartPar
ECS (\sphinxhref{https://doi.org/10.1093/bioinformatics/bty682}{the ECS paper}) first converts the p\sphinxhyphen{}values of a gene to chi\sphinxhyphen{}square statistics(one degree of freedom). Then, merges all chi\sphinxhyphen{}square statistics of a gene after correcting the redundancy of the statistics due to LD. The merged statistic is called an ECS which is used to calculate the p\sphinxhyphen{}value of the gene.


\subsection{Synopsis}
\label{\detokenize{detailed_document:synopsis}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xms16g \PYGZhy{}Xmx16g \PYGZhy{}jar kggsee.jar
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}assoc
  \PYGZhy{}\PYGZhy{}out \PYGZlt{}prefix\PYGZgt{}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}chrom\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: CHR}
  \PYGZhy{}\PYGZhy{}pos\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: BP}
  \PYGZhy{}\PYGZhy{}p\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: P}
  \PYGZhy{}\PYGZhy{}neargene \PYGZlt{}basepair\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 5000}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}filter\PYGZhy{}eqtl\PYGZhy{}p \PYGZlt{}pval\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 0.01}
\end{sphinxVerbatim}

\sphinxAtStartPar
The flag \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}assoc}} triggers the gene\sphinxhyphen{}based association tests. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}} specifies a white space\sphinxhyphen{}delimited GWAS summary statistic file which must have three columns of the chromosome of SNP, coordinate of SNP, and p\sphinxhyphen{}value of SNP; headers of the three columns can be specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}chrom\sphinxhyphen{}col}}, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}pos\sphinxhyphen{}col}} and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}p\sphinxhyphen{}col}} separately. SNPs belonging to a gene can be defined either by SNPs close to the gene or by eQTLs of the gene. If \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}} is specified, KGGSEE reads gene annotations and considers SNPs inside a gene and its adjacent regions at a fixed number of basepairs on both sides to be a test unit. If \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}} is specified, KGGSEE reads the eQTL summary statistic file and considers eQTLs of a gene or a transcript to be a test unit, and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}} is overridden. When \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}} is specified, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}eqtl\sphinxhyphen{}p}} can be used to specify a threshold of eQTL p\sphinxhyphen{}values. Only eQTLs with a p\sphinxhyphen{}value lower than the threshold will be considered. {\hyperref[\detokenize{detailed_document:eqtl-file}]{\sphinxcrossref{\DUrole{std,std-ref}{A description of the eQTL file format}}}} is near the beginning of this chapter.


\newpage
\subsection{Examples}
\label{\detokenize{detailed_document:examples}}

\subsubsection{Gene\sphinxhyphen{}based association tests based on physical distance}
\label{\detokenize{detailed_document:gene-based-association-tests-based-on-physical-distance}}
\sphinxAtStartPar
In this example, SNPs inside a gene and its 10 kb adjacent regions will be grouped for association tests.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
\PYGZhy{}\PYGZhy{}gene\PYGZhy{}assoc \PYG{l+s+se}{\PYGZbs{}}
\PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
\PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
\PYGZhy{}\PYGZhy{}neargene 10000 \PYG{l+s+se}{\PYGZbs{}}
\PYGZhy{}\PYGZhy{}out t1.1
\end{sphinxVerbatim}


\subsubsection{Gene\sphinxhyphen{}based association tests based on eQTLs}
\label{\detokenize{detailed_document:gene-based-association-tests-based-on-eqtls}}
\sphinxAtStartPar
In this example, eQTLs of a gene will be grouped for association tests.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}assoc \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}gene\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t1.2
\end{sphinxVerbatim}


\subsubsection{Transcript\sphinxhyphen{}based association tests based on eQTLs}
\label{\detokenize{detailed_document:transcript-based-association-tests-based-on-eqtls}}
\sphinxAtStartPar
In this example, eQTLs of a transcript will be grouped for association tests.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}assoc \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}transcript\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t1.3
\end{sphinxVerbatim}


\subsection{Outputs}
\label{\detokenize{detailed_document:outputs}}
\sphinxAtStartPar
The file with a suffix of \sphinxcode{\sphinxupquote{.gene.pvalue.txt}} saves the results of gene\sphinxhyphen{}based association tests. Columns of the file are as follow:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Gene
&
\sphinxAtStartPar
Gene symbol
\\
\hline
\sphinxAtStartPar
\#Var
&
\sphinxAtStartPar
Number of variants within the gene
\\
\hline
\sphinxAtStartPar
ECSP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of ECS
\\
\hline
\sphinxAtStartPar
GATESP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of GATES
\\
\hline
\sphinxAtStartPar
Chrom
&
\sphinxAtStartPar
Chromosome of the gene
\\
\hline
\sphinxAtStartPar
Pos
&
\sphinxAtStartPar
The coordinate of the variant with the lowest p\sphinxhyphen{}value within the gene
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_P
&
\sphinxAtStartPar
p\sphinxhyphen{}value of the variant
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
Columns of the file with the suffix of \sphinxcode{\sphinxupquote{.gene.var.pvalue.txt.gz}} are the same as \sphinxcode{\sphinxupquote{*.gene.pvalue.txt}}. The difference is that, for each gene, in \sphinxcode{\sphinxupquote{*.gene.pvalue.txt}}, only the variant with the lowest p\sphinxhyphen{}value is output, while in \sphinxcode{\sphinxupquote{*.gene.var.pvalue.txt.gz}}, all variants are output. The file with the suffix of \sphinxcode{\sphinxupquote{.qq.png}} is the Q\sphinxhyphen{}Q plots for p\sphinxhyphen{}values of GWAS summary statistics and gene\sphinxhyphen{}based association tests by GATES and ECS.


\section{DESE}
\label{\detokenize{detailed_document:dese}}\label{\detokenize{detailed_document:detail-dese}}
\sphinxAtStartPar
DESE (\sphinxstylestrong{D}river\sphinxhyphen{}tissue \sphinxstylestrong{E}stimation by \sphinxstylestrong{S}elective \sphinxstylestrong{E}xpression; \sphinxhref{https://doi.org/10.1186/s13059-019-1801-5}{the DESE paper}) estimates driver tissues by tissue\sphinxhyphen{}selective expression of phenotype\sphinxhyphen{}associated genes in GWAS. The assumption is that the tissue\sphinxhyphen{}selective expression of causal or susceptibility genes indicates the tissues where complex phenotypes happen primarily, which are called driver tissues. Therefore, a driver tissue is very likely to be enriched with selective expression of susceptibility genes of a phenotype.

\sphinxAtStartPar
DESE initially performed the association analysis by mapping SNPs to genes according to their physical distance. We further demonstrated that grouping eQTLs of a gene or a transcript to perform the association analysis could be more powerful. We named the \sphinxstylestrong{e}QTL\sphinxhyphen{}guided \sphinxstylestrong{DESE} eDESE. KGGSEE implements DESE and eDESE with an improved effective chi\sphinxhyphen{}squared statistic to control type I error rates and remove redundant associations (\sphinxhref{https://doi.org/10.7554/eLife.70779}{the eDESE paper}).


\subsection{Synopsis}
\label{\detokenize{detailed_document:id2}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xms16g \PYGZhy{}Xmx16g \PYGZhy{}jar kggsee.jar
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}finemapping
  \PYGZhy{}\PYGZhy{}out \PYGZlt{}prefix\PYGZgt{}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}chrom\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: CHR}
  \PYGZhy{}\PYGZhy{}pos\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: BP}
  \PYGZhy{}\PYGZhy{}p\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: P}
  \PYGZhy{}\PYGZhy{}neargene \PYGZlt{}basepair\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 5000}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}filter\PYGZhy{}eqtl\PYGZhy{}p \PYGZlt{}pval\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 0.01}
  \PYGZhy{}\PYGZhy{}multiple\PYGZhy{}testing \PYGZlt{}bonf\PYG{p}{|}benfdr\PYG{p}{|}fixed\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: bonf}
  \PYGZhy{}\PYGZhy{}p\PYGZhy{}value\PYGZhy{}cutoff \PYGZlt{}pval\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 0.05}
  \PYGZhy{}\PYGZhy{}top\PYGZhy{}gene \PYGZlt{}number\PYGZgt{}
  \PYGZhy{}\PYGZhy{}expression\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}geneset\PYGZhy{}db \PYGZlt{}cura\PYG{p}{|}cgp\PYG{p}{|}cano\PYG{p}{|}cmop\PYG{p}{|}onto\PYG{p}{|}onco\PYG{p}{|}immu\PYGZgt{}
  \PYGZhy{}\PYGZhy{}geneset\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
\end{sphinxVerbatim}

\sphinxAtStartPar
The flag \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}finemapping}} triggers DESE. First, KGGSEE performs gene\sphinxhyphen{}based association tests, which is the same as the analyses triggered by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}assoc}}. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}} specifies a white space delimited GWAS summary statistic file which must have three columns of the chromosome of SNP, coordinate of SNP, and p\sphinxhyphen{}value of SNP; headers of the three columns can be specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}chrom\sphinxhyphen{}col}}, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}pos\sphinxhyphen{}col}} and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}p\sphinxhyphen{}col}} separately. SNPs belonging to a gene can be defined either by SNPs close to the gene or by eQTLs of the gene. If \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}} is specified, KGGSEE reads gene annotations and considers SNPs inside a gene and its adjacent regions at a fixed number of basepairs on both sides to be a test unit. If \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}} is specified, eDESE is evoked; KGGSEE reads the eQTL summary statistic file and considers eQTLs of a gene or a transcript to be a test unit, and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}} is overridden. When \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}} is specified, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}eqtl\sphinxhyphen{}p}} can be used to specify a threshold of eQTL p\sphinxhyphen{}values. Only eQTLs with a p\sphinxhyphen{}value lower than the threshold will be considered. {\hyperref[\detokenize{detailed_document:eqtl-file}]{\sphinxcrossref{\DUrole{std,std-ref}{A description of the eQTL file format}}}} is near the beginning of this chapter.

\sphinxAtStartPar
Second, after the gene\sphinxhyphen{}based association tests, significant genes by ECS are retained for fine\sphinxhyphen{}mapping. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}multiple\sphinxhyphen{}testing}} specifies the method for multiple testing correction: \sphinxcode{\sphinxupquote{bonf}} denotes Bonferroni correction; \sphinxcode{\sphinxupquote{benfdr}} denotes Benjamini\textendash{}Hochberg FDR; \sphinxcode{\sphinxupquote{fixed}} denotes no correction. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}p\sphinxhyphen{}value\sphinxhyphen{}cutoff}} specifies the threshold of the adjusted p\sphinxhyphen{}value. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}top\sphinxhyphen{}gene}} specifies the maximum number of genes retained for fine\sphinxhyphen{}mapping. So, only genes (no more than the specified maximum number) with adjusted p\sphinxhyphen{}values lower than the specified threshold are retained for fine\sphinxhyphen{}mapping. Then, KGGSEE reads the expression file specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}expression\sphinxhyphen{}file}} and performs iterative estimation of driver tissues.

\sphinxAtStartPar
Finally, if \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}geneset\sphinxhyphen{}db}} is specified, KGGSEE tests if the conditional significant genes are enriched in gene sets of \sphinxhref{http://www.gsea-msigdb.org/gsea/msigdb/index.jsp}{MSigDB}. The abbreviations of gene sets are as follow:
\begin{quote}

\begin{DUlineblock}{0em}
\item[] \sphinxcode{\sphinxupquote{cura}}: C2. curated gene sets;
\item[] \sphinxcode{\sphinxupquote{cgp}} : C2. chemical and genetic perturbations;
\item[] \sphinxcode{\sphinxupquote{cano}}: C2. canonical pathways;
\item[] \sphinxcode{\sphinxupquote{cmop}}: C4. computational gene sets;
\item[] \sphinxcode{\sphinxupquote{onto}}: C5. ontology gene sets;
\item[] \sphinxcode{\sphinxupquote{onco}}: C6. oncogenic signature gene sets;
\item[] \sphinxcode{\sphinxupquote{immu}}: C7. immunologic signature gene sets.
\end{DUlineblock}
\end{quote}

\sphinxAtStartPar
Customized gene sets for enrichment tests can be specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}geneset\sphinxhyphen{}file}}. Please refer to \sphinxcode{\sphinxupquote{resources/*.symbols.gmt.gz}} under the KGGSEE directory for file formats.

\sphinxAtStartPar
Expression files should be white space delimitated. The first column is gene/transcript IDs. The following columns are means and standard errors of expression levels of genes or transcripts in multiple tissues. A gene\sphinxhyphen{}level expression file looks like this:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Name                 Tissue1.mean   Tissue1.SE     Tissue2.mean   Tissue2.SE     ...
ENSG00000223972.5    0.0038016      0.00036668     0.0045709      0.00046303     ...
ENSG00000227232.5    1.9911         0.030021       1.8841         0.040247       ...
ENSG00000278267.1    0.00049215     0.00010645     0.00036466     9.2944E-05     ...
ENSG00000243485.5    0.0047772      0.00038018     0.0067897      0.00074318     ...
ENSG00000237613.2    0.0030462      0.00027513     0.0030465      0.00031694     ...
ENSG00000268020.3    0.011766       0.00061769     0.013409       0.0011429      ...
ENSG00000240361.1    0.017913       0.00093294     0.021833       0.001556       ...
\end{sphinxVerbatim}

\sphinxAtStartPar
A transcript\sphinxhyphen{}level expression file looks like this:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Name                                   Tissue1.mean   Tissue1.SE     ...
ENST00000373020.8:ENSG00000000003.14   35.06          0.52271        ...
ENST00000494424.1:ENSG00000000003.14   0.0034329      0.001209       ...
ENST00000496771.5:ENSG00000000003.14   1.0462         0.019697       ...
ENST00000612152.4:ENSG00000000003.14   2.5764         0.041124       ...
ENST00000614008.4:ENSG00000000003.14   0.42826        0.01346        ...
ENST00000373031.4:ENSG00000000005.5    15.215         0.58333        ...
ENST00000485971.1:ENSG00000000005.5    1.0715         0.04074        ...
\end{sphinxVerbatim}


\newpage
\subsection{Examples}
\label{\detokenize{detailed_document:id3}}

\subsubsection{DESE based on physical distance}
\label{\detokenize{detailed_document:dese-based-on-physical-distance}}
\sphinxAtStartPar
In this example, SNPs inside a gene and its 10 kb adjacent regions will be considered as belonging to a gene. Significant genes by ECS with FDR\textless{}0.05 will be retained for fine\sphinxhyphen{}mapping.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}finemapping \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}neargene 10000 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}multiple\PYGZhy{}testing benfdr \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}p\PYGZhy{}value\PYGZhy{}cutoff 0.05 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}expression\PYGZhy{}file GTEx_v8_TMM.gene.meanSE.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t2.1
\end{sphinxVerbatim}


\subsubsection{eDESE based on gene\sphinxhyphen{}level eQTLs}
\label{\detokenize{detailed_document:edese-based-on-gene-level-eqtls}}
\sphinxAtStartPar
In this example, eQTLs of a gene will be considered as a unit for a gene\sphinxhyphen{}based association test. The top 100 significant genes by ECS with nominal p\textless{}0.05 will be retained for fine\sphinxhyphen{}mapping. Significant genes by eDESE will be tested if they are enriched in the C5. ontology gene sets of \sphinxhref{http://www.gsea-msigdb.org/gsea/msigdb/index.jsp}{MSigDB}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}finemapping \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}gene\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}multiple\PYGZhy{}testing fixed \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}p\PYGZhy{}value\PYGZhy{}cutoff 0.05 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}top\PYGZhy{}gene 100 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}expression\PYGZhy{}file GTEx_v8_TMM.gene.meanSE.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}geneset\PYGZhy{}db onto \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t2.2
\end{sphinxVerbatim}


\subsubsection{SelDP based on gene\sphinxhyphen{}level eQTLs}
\label{\detokenize{detailed_document:seldp-based-on-gene-level-eqtls}}
\sphinxAtStartPar
In this example, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}expression\sphinxhyphen{}file}} specifies a customized file of the drug\sphinxhyphen{}induced gene\sphinxhyphen{}expression fold\sphinxhyphen{}change profile which has the same format as a gene expression file. SelDP estimates the drug selective perturbation effect on the phenotype\sphinxhyphen{}associated genes’ expression to aid the drug repositioning for complex diseases.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}gene\PYGZhy{}finemapping \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}genet\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}expression\PYGZhy{}file drug\PYGZhy{}induced\PYGZus{}expression\PYGZus{}change\PYGZus{}profile \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t2.3
\end{sphinxVerbatim}


\subsection{Outputs}
\label{\detokenize{detailed_document:id5}}
\sphinxAtStartPar
The three files with suffixes of \sphinxcode{\sphinxupquote{.gene.pvalue.txt}}, \sphinxcode{\sphinxupquote{.gene.var.pvalue.txt.gz}}, and \sphinxcode{\sphinxupquote{.qq.png}} are the same as their counterparts output by {\hyperref[\detokenize{detailed_document:detail-ecs}]{\sphinxcrossref{\DUrole{std,std-ref}{Gene\sphinxhyphen{}based association tests}}}}.

\sphinxAtStartPar
In addition, results of conditional gene\sphinxhyphen{}based association tests are saved in a file with a suffix of \sphinxcode{\sphinxupquote{.finemapping.gene.ecs.txt}}. Columns of the file are as follow:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Gene
&
\sphinxAtStartPar
Gene symbol
\\
\hline
\sphinxAtStartPar
Chrom
&
\sphinxAtStartPar
Chromosome of the gene
\\
\hline
\sphinxAtStartPar
StartPos
&
\sphinxAtStartPar
Start position of the gene
\\
\hline
\sphinxAtStartPar
EndPos
&
\sphinxAtStartPar
End position of the gene
\\
\hline
\sphinxAtStartPar
\#Var
&
\sphinxAtStartPar
Number of variants within the gene
\\
\hline
\sphinxAtStartPar
Group
&
\sphinxAtStartPar
LD group number. Conditional ECS tests were performed for genes within the same LD group.
\\
\hline
\sphinxAtStartPar
ECSP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of ECS
\\
\hline
\sphinxAtStartPar
CondiECSP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of conditional gene\sphinxhyphen{}based association tests by conditional ECS
\\
\hline
\sphinxAtStartPar
GeneScore
&
\sphinxAtStartPar
The gene’s selective expression score in all tissues. A gene with a high score will be given higher priority to enter the conditioning procedure.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
Results of phenotype\sphinxhyphen{}tissue associations are saved in a file with a suffix of \sphinxcode{\sphinxupquote{.celltype.txt}}. Columns of the file are as follow:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
TissueName
&
\sphinxAtStartPar
Name of the tissue being tested
\\
\hline
\sphinxAtStartPar
PValue
&
\sphinxAtStartPar
p\sphinxhyphen{}values of phenotype\sphinxhyphen{}tissue associations. This is basically a Wilcoxon rank\sphinxhyphen{}sum test which tests whether the selective expression median of the phenotype\sphinxhyphen{}associated genes is significantly higher than that of other genes in an interrogated tissue.
\\
\hline
\sphinxAtStartPar
Log(p)
&
\sphinxAtStartPar
The negative logarithm (base 10) of p\sphinxhyphen{}values of phenotype\sphinxhyphen{}tissue association
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
If \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}geneset\sphinxhyphen{}db}} or \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}geneset\sphinxhyphen{}file}} is specified, results of enrichment tests are saved in a file with a suffix of \sphinxcode{\sphinxupquote{.geneset.txt}}. Columns of the file are as follow:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{4}{11}|\X{7}{11}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
GeneSet\_ID
&
\sphinxAtStartPar
Gene\sphinxhyphen{}set ID in the first column of the gene\sphinxhyphen{}set file
\\
\hline
\sphinxAtStartPar
Enrichment\_PValue\_Hypergeometric
&
\sphinxAtStartPar
p\sphinxhyphen{}values of the hypergeometric tests.
\\
\hline
\sphinxAtStartPar
IsSignificant\_Hypergeometric
&
\sphinxAtStartPar
If the conditional significant genes are significantly enriched in the gene set.
\\
\hline
\sphinxAtStartPar
Total\_GeneSet\_Gene\#
&
\sphinxAtStartPar
The total number of genes in the gene set.
\\
\hline
\sphinxAtStartPar
GeneSet\_URL
&
\sphinxAtStartPar
Gene\sphinxhyphen{}set URL in the second column of the gene\sphinxhyphen{}set file
\\
\hline
\sphinxAtStartPar
Gene\_PValue
&
\sphinxAtStartPar
p\sphinxhyphen{}values of conditional significant genes within the gene set.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\section{EMIC}
\label{\detokenize{detailed_document:emic}}\label{\detokenize{detailed_document:detail-emic}}
\sphinxAtStartPar
EMIC (\sphinxstylestrong{E}ffective\sphinxhyphen{}median\sphinxhyphen{}based \sphinxstylestrong{M}endelian randomization framework for \sphinxstylestrong{I}nferring the \sphinxstylestrong{C}ausal genes of complex phenotypes) inferences gene expressions’ causal effect on a complex phenotype with dependent expression quantitative loci by a robust median\sphinxhyphen{}based Mendelian randomization. The effective\sphinxhyphen{}median method solved the high false\sphinxhyphen{}positive issue in the existing MR methods due to either correlation among instrumental variables or noises in approximated linkage disequilibrium (LD). EMIC can further perform a pleiotropy fine\sphinxhyphen{}mapping analysis to remove possible false\sphinxhyphen{}positive estimates (\sphinxhref{https://doi.org/10.1016/j.ajhg.2022.04.004}{the EMIC paper}).


\subsection{Synopsis}
\label{\detokenize{detailed_document:id6}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xms16g \PYGZhy{}Xmx16g \PYGZhy{}jar kggsee.jar
  \PYGZhy{}\PYGZhy{}emic
  \PYGZhy{}\PYGZhy{}out \PYGZlt{}prefix\PYGZgt{}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}chrom\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: CHR}
  \PYGZhy{}\PYGZhy{}pos\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: BP}
  \PYGZhy{}\PYGZhy{}a1\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: A1}
  \PYGZhy{}\PYGZhy{}a2\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: A2}
  \PYGZhy{}\PYGZhy{}freq\PYGZhy{}a1\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: FRQ\PYGZus{}U}
  \PYGZhy{}\PYGZhy{}beta\PYGZhy{}col \PYGZlt{}header\PYGZgt{}
  \PYGZhy{}\PYGZhy{}beta\PYGZhy{}type \PYGZlt{}0\PYG{p}{|}1\PYG{p}{|}2\PYGZgt{}
  \PYGZhy{}\PYGZhy{}se\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: SE}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}filter\PYGZhy{}eqtl\PYGZhy{}p \PYGZlt{}pval\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 1E\PYGZhy{}4}
  \PYGZhy{}\PYGZhy{}ld\PYGZhy{}pruning\PYGZhy{}mr  \PYGZlt{}r2\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 0.5}
  \PYGZhy{}\PYGZhy{}emic\PYGZhy{}pfm\PYGZhy{}p \PYGZlt{}pval\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 2.5E\PYGZhy{}6}
  \PYGZhy{}\PYGZhy{}emic\PYGZhy{}plot\PYGZhy{}p \PYGZlt{}pval\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 2.5E\PYGZhy{}3}
\end{sphinxVerbatim}

\sphinxAtStartPar
When performing EMIC (triggered by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic}}), a GWAS summary statistic file (specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}}) and an eQTL summary statistic file (specified by \sphinxcode{\sphinxupquote{eqtl\sphinxhyphen{}file}}) are needed. The GWAS summary statistic file must have columns of SNP coordinates (specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}chrom\sphinxhyphen{}col}} and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}pos\sphinxhyphen{}col}}), the two alleles (specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}a1\sphinxhyphen{}col}} and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}a2\sphinxhyphen{}col}}), frequencies of the allele specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}a1\sphinxhyphen{}col}} (specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}freq\sphinxhyphen{}a1\sphinxhyphen{}col}}), the effect sizes and its standard errors (specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}beta\sphinxhyphen{}col}} and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}se\sphinxhyphen{}col}}). The type of effect sizes is specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}beta\sphinxhyphen{}type}} (\sphinxcode{\sphinxupquote{0}} for linear regression coefficient of a quantitative phenotype; \sphinxcode{\sphinxupquote{1}} for the logarithm of odds ratio or logistic regression coefficient of a qualitative phenotype; \sphinxcode{\sphinxupquote{2}} for an odds ratio of a qualitative phenotype). \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}eqtl\sphinxhyphen{}p}} specifies the p\sphinxhyphen{}value threshold of eQTLs; only eQTLs with a p\sphinxhyphen{}value lower than the threshold will be considered; we note here that the default value is \sphinxcode{\sphinxupquote{1E\sphinxhyphen{}4}} for EMIC, which is different from the other analyses. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}ld\sphinxhyphen{}pruning\sphinxhyphen{}mr}} specifies the threshold of LD coefficient when pruning variants; for each gene or transcript, eQTLs with LD coefficients higher than the threshold will be pruned. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic\sphinxhyphen{}pfm\sphinxhyphen{}p}} specifies the p\sphinxhyphen{}value threshold to further perform an EMIC pleiotropy fine\sphinxhyphen{}mapping (EMIC\sphinxhyphen{}PFM) analysis; if the EMIC p\sphinxhyphen{}value of a gene is lower than the threshold, an EMIC\sphinxhyphen{}PFM will be performed to control the false\sphinxhyphen{}positive caused by pleiotropy. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic\sphinxhyphen{}plot\sphinxhyphen{}p}} specifies the p\sphinxhyphen{}value threshold for plotting a scatter plot; genes with an EMIC p\sphinxhyphen{}value lower than the threshold will be plotted. {\hyperref[\detokenize{detailed_document:eqtl-file}]{\sphinxcrossref{\DUrole{std,std-ref}{A description of the eQTL file format}}}} is near the beginning of this chapter.


\subsection{Examples}
\label{\detokenize{detailed_document:id7}}

\subsubsection{EMIC based on gene\sphinxhyphen{}level eQTL}
\label{\detokenize{detailed_document:emic-based-on-gene-level-eqtl}}
\sphinxAtStartPar
This is an example of gene\sphinxhyphen{}level EMIC. Only eQTLs with a p\sphinxhyphen{}value lower than 1E\sphinxhyphen{}6 will be considered IVs. Genes with a p\sphinxhyphen{}value of EMIC lower than 0.05 will also undergo EMIC\sphinxhyphen{}PFM. Genes with a p\sphinxhyphen{}value of EMIC lower than 0.01 will be plotted.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}gene\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}beta\PYGZhy{}col OR \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}beta\PYGZhy{}type 2 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}emic \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}filter\PYGZhy{}eqtl\PYGZhy{}p 1e\PYGZhy{}6 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}emic\PYGZhy{}pfm\PYGZhy{}p 0.05 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}emic\PYGZhy{}plot\PYGZhy{}p 0.01 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t3.1
\end{sphinxVerbatim}


\subsubsection{EMIC based on transcript\sphinxhyphen{}level eQTL}
\label{\detokenize{detailed_document:emic-based-on-transcript-level-eqtl}}
\sphinxAtStartPar
This is an example of transcript\sphinxhyphen{}level EMIC. Only eQTLs with a p\sphinxhyphen{}value lower than 1E\sphinxhyphen{}6 will be considered IVs. Transcripts with a p\sphinxhyphen{}value of EMIC lower than 0.05 will also undergo EMIC\sphinxhyphen{}PFM. Transcripts with a p\sphinxhyphen{}value of EMIC lower than 0.01 will be plotted.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}transcript\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}beta\PYGZhy{}col OR \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}beta\PYGZhy{}type 2 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}emic \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}filter\PYGZhy{}eqtl\PYGZhy{}p 1e\PYGZhy{}6 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}emic\PYGZhy{}pfm\PYGZhy{}p 0.05 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}emic\PYGZhy{}plot\PYGZhy{}p 0.01 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t3.2
\end{sphinxVerbatim}


\subsection{Outputs}
\label{\detokenize{detailed_document:id8}}
\sphinxAtStartPar
The numeric results of EMIC are saved in a file with a suffix of \sphinxcode{\sphinxupquote{.emic.gene.txt}}. There are nine columns in the file:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Gene
&
\sphinxAtStartPar
The gene symbol
\\
\hline
\sphinxAtStartPar
\#Var
&
\sphinxAtStartPar
Number of IVs within the gene
\\
\hline
\sphinxAtStartPar
minP\_EMIC
&
\sphinxAtStartPar
p\sphinxhyphen{}value of EMIC. When a transcript\sphinxhyphen{}level EMIC is performed, this is the minimum p\sphinxhyphen{}value among all transcripts of the gene.
\\
\hline
\sphinxAtStartPar
Details\_EMIC
&
\sphinxAtStartPar
Detailed results of EMIC\sphinxhyphen{}PFM separated by semicolons. Each result has four components in brackets: the number of IVs, the causal effect estimate and its standard error, and the p\sphinxhyphen{}value. When a transcript\sphinxhyphen{}level EMIC is performed, results for each transcript are listed.
\\
\hline
\sphinxAtStartPar
Chrom
&
\sphinxAtStartPar
Chromosome of the gene
\\
\hline
\sphinxAtStartPar
Pos
&
\sphinxAtStartPar
The coordinate of the IV with the lowest GWAS p\sphinxhyphen{}value
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_P
&
\sphinxAtStartPar
GWAS p\sphinxhyphen{}value of the IV
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_Beta
&
\sphinxAtStartPar
The phenotype association effect size of the IV
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_SE
&
\sphinxAtStartPar
Standard error of the effect size
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
The numeric results of EMIC\sphinxhyphen{}PFM are saved in a file with a suffix of \sphinxcode{\sphinxupquote{.emic.gene.PleiotropyFinemapping.txt}}. Only genes with a p\sphinxhyphen{}value lower than the threshold specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic\sphinxhyphen{}pfm\sphinxhyphen{}p}} are saved. The file has thirteen columns, in which nine are the same as columns of \sphinxcode{\sphinxupquote{*.emic.gene.txt}}. The other four columns are:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Group
&
\sphinxAtStartPar
IDs of a group of genes that share eQTLs.
\\
\hline
\sphinxAtStartPar
minP\_EMIC\_PFM
&
\sphinxAtStartPar
p\sphinxhyphen{}value of EMIC\sphinxhyphen{}PFM. When a transcript\sphinxhyphen{}level EMIC\sphinxhyphen{}PFM is performed, this is the minimum p\sphinxhyphen{}value among all transcripts of the gene.
\\
\hline
\sphinxAtStartPar
DetailsEMIC\_PFM
&
\sphinxAtStartPar
Detailed results of EMIC\sphinxhyphen{}PFM separated by semicolons. Each result has four components in brackets: the number of IVs, the causal effect estimate and its standard error, and the p\sphinxhyphen{}value. When a transcript\sphinxhyphen{}level EMIC\sphinxhyphen{}PFM is performed, results for each transcript are listed.
\\
\hline
\sphinxAtStartPar
CochransQ
&
\sphinxAtStartPar
The p\sphinxhyphen{}value of an extended Cochran’s Q test. The significance (p\textless{}1E\sphinxhyphen{}3) means that the causal effect is more likely to be false\sphinxhyphen{}positive. At this point, KGGSEE excludes its eQTLs which are also the eQTLs of other significant genes, and redoes EMIC. In this case, results in the columns of minP\_EMIC\_PFM and DetailsEMIC\_PFM will be different from in the columns of minP\_EMIC and Details\_EMIC.
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
Columns of the file with a suffix of \sphinxcode{\sphinxupquote{.emic.gene.var.tsv.gz}} are the same as \sphinxcode{\sphinxupquote{*.emic.gene.txt}}. The difference is that, for each gene, in \sphinxcode{\sphinxupquote{*.emic.gene.txt}}, only the eQTL with the lowest GWAS p\sphinxhyphen{}value is output, while in \sphinxcode{\sphinxupquote{*.emic.gene.var.tsv.gz}}, all eQTLs are output. The file with a suffix of \sphinxcode{\sphinxupquote{.qq.png}} saves the Q\sphinxhyphen{}Q plot for GWAS p\sphinxhyphen{}values of IVs. The file with a suffix of \sphinxcode{\sphinxupquote{.emic.qq.png}} saves the Q\sphinxhyphen{}Q plot for EMIC p\sphinxhyphen{}values. The file with a suffix of \sphinxcode{\sphinxupquote{.scatterplots.emic.pdf}} saves the scatter plots of genetic association with gene expression. Each gene with an EMIC p\sphinxhyphen{}value lower than the threshold specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic\sphinxhyphen{}plot\sphinxhyphen{}p}} is saved on a separate page of the PDF. A filled rectangle on the plots denotes an IV. The red rectangle denotes the most significant GWAS variant among all the IVs of a gene. The slope of the line represents the estimated causal effect. The color of an IV denotes the degree of the LD between the IV and the most significant GWAS variant. The error bars in the rectangles denote the standard errors of the coefficient estimates.


\section{Gene\sphinxhyphen{}based heritability estimation}
\label{\detokenize{detailed_document:gene-based-heritability-estimation}}\label{\detokenize{detailed_document:detail-h2}}
\sphinxAtStartPar
This analysis estimates the heritability of each gene and performs gene\sphinxhyphen{}based association tests at the same time.


\subsection{Synopsis}
\label{\detokenize{detailed_document:id9}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xms16g \PYGZhy{}Xmx16g \PYGZhy{}jar kggsee.jar
  \PYGZhy{}\PYGZhy{}estimate\PYGZhy{}heritability
  \PYGZhy{}\PYGZhy{}out \PYGZlt{}prefix\PYGZgt{}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}chrom\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: CHR}
  \PYGZhy{}\PYGZhy{}pos\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: BP}
  \PYGZhy{}\PYGZhy{}p\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: P}
  \PYGZhy{}\PYGZhy{}nmiss\PYGZhy{}col \PYGZlt{}header\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: Neff}
  \PYGZhy{}\PYGZhy{}case\PYGZhy{}col \PYGZlt{}header\PYGZgt{}
  \PYGZhy{}\PYGZhy{}control\PYGZhy{}col \PYGZlt{}header\PYGZgt{}
  \PYGZhy{}\PYGZhy{}neargene \PYGZlt{}basepair\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 5000}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file \PYGZlt{}file\PYGZgt{}
  \PYGZhy{}\PYGZhy{}filter\PYGZhy{}eqtl\PYGZhy{}p \PYGZlt{}pval\PYGZgt{}  \PYG{c+c1}{\PYGZsh{} default: 0.01}
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}estimate\sphinxhyphen{}heritability}} triggers gene\sphinxhyphen{}based association tests and estimation of gene heritability. \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}} specifies a white space delimited GWAS summary statistic file which must have three columns of the chromosome of SNP, coordinate of SNP, and p\sphinxhyphen{}value of SNP; headers of the three columns can be specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}chrom\sphinxhyphen{}col}}, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}pos\sphinxhyphen{}col}} and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}p\sphinxhyphen{}col}} separately. In addition, for quantitative phenotype, a column of sample sizes is needed, and its header is specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}nmiss\sphinxhyphen{}col}}; for qualitative phenotype, two columns of case sample sizes and control sample sizes are needed, and their header is specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}case\sphinxhyphen{}col}} and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}control\sphinxhyphen{}col}} separately. SNPs belonging to a gene can be defined either by SNPs close to the gene or by eQTLs of the gene. If \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}} is specified, KGGSEE reads gene annotations and considers SNPs inside a gene and its adjacent regions at a fixed number of basepairs on both sides to be a test unit. If \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}} is specified, KGGSEE reads the eQTL summary statistic file and considers eQTLs of a gene or a transcript to be a test unit, and \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}} is overridden. When \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}} is specified, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}eqtl\sphinxhyphen{}p}} can be used to specify a threshold of eQTL p\sphinxhyphen{}values. Only eQTLs with a p\sphinxhyphen{}value lower than the threshold will be considered. {\hyperref[\detokenize{detailed_document:eqtl-file}]{\sphinxcrossref{\DUrole{std,std-ref}{A description of the eQTL file format}}}} is near the beginning of this chapter.


\subsection{Examples}
\label{\detokenize{detailed_document:id10}}

\subsubsection{Gene heritability based on physical distance}
\label{\detokenize{detailed_document:gene-heritability-based-on-physical-distance}}
\sphinxAtStartPar
In this example, SNPs inside a gene and its 10 kb adjacent regions will be grouped to estimate heritability.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}estimate\PYGZhy{}heritability \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}case\PYGZhy{}col Nca \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}control\PYGZhy{}col Nco \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}neargene 10000 \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t4.1
\end{sphinxVerbatim}


\subsubsection{Gene heritability based on eQTLs}
\label{\detokenize{detailed_document:gene-heritability-based-on-eqtls}}
\sphinxAtStartPar
In this example, eQTLs of a gene will be grouped to estimate heritability.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}estimate\PYGZhy{}heritability \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}case\PYGZhy{}col Nca \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}control\PYGZhy{}col Nco \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}gene\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t4.2
\end{sphinxVerbatim}


\subsubsection{Transcript heritability based on eQTLs}
\label{\detokenize{detailed_document:transcript-heritability-based-on-eqtls}}
\sphinxAtStartPar
In this example, eQTLs of a transcript will be grouped to estimate heritability.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
java \PYGZhy{}Xmx4g \PYGZhy{}jar ../kggsee.jar \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}estimate\PYGZhy{}heritability \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}vcf\PYGZhy{}ref 1kg\PYGZus{}hg19\PYGZus{}eur\PYGZus{}chr1.vcf.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}sum\PYGZhy{}file scz\PYGZus{}gwas\PYGZus{}eur\PYGZus{}chr1.tsv.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}case\PYGZhy{}col Nca \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}control\PYGZhy{}col Nco \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}eqtl\PYGZhy{}file GTEx\PYGZus{}v8\PYGZus{}transcript\PYGZus{}BrainBA9.eqtl.txt.gz \PYG{l+s+se}{\PYGZbs{}}
  \PYGZhy{}\PYGZhy{}out t4.3
\end{sphinxVerbatim}


\subsection{Outputs}
\label{\detokenize{detailed_document:id11}}
\sphinxAtStartPar
The file with a suffix of \sphinxcode{\sphinxupquote{.gene.pvalue.txt}} saves the results of gene\sphinxhyphen{}based heritability estimates and association tests. Columns of the file are as follow:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{1}{5}|\X{4}{5}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Header
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
\\
\hline
\sphinxAtStartPar
Gene
&
\sphinxAtStartPar
Gene symbol
\\
\hline
\sphinxAtStartPar
\#Var
&
\sphinxAtStartPar
Number of variants within the gene
\\
\hline
\sphinxAtStartPar
ECSP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of ECS
\\
\hline
\sphinxAtStartPar
GATESP
&
\sphinxAtStartPar
p\sphinxhyphen{}value of GATES
\\
\hline
\sphinxAtStartPar
Herit
&
\sphinxAtStartPar
Heritability estimate
\\
\hline
\sphinxAtStartPar
HeritSE
&
\sphinxAtStartPar
Standard error of the heritability estimate
\\
\hline
\sphinxAtStartPar
Chrom
&
\sphinxAtStartPar
Chromosome of the gene
\\
\hline
\sphinxAtStartPar
Pos
&
\sphinxAtStartPar
The coordinate of the variant with the lowest p\sphinxhyphen{}value within the gene
\\
\hline
\sphinxAtStartPar
GWAS\_Var\_P
&
\sphinxAtStartPar
p\sphinxhyphen{}value of the variant
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
Columns of the file with the suffix of \sphinxcode{\sphinxupquote{.gene.var.pvalue.txt.gz}} are the same as \sphinxcode{\sphinxupquote{*.gene.pvalue.txt}}. The difference is that, for each gene, in \sphinxcode{\sphinxupquote{*.gene.pvalue.txt}}, only the variant with the lowest p\sphinxhyphen{}value is output, while in \sphinxcode{\sphinxupquote{*.gene.var.pvalue.txt.gz}}, all variants are output. The file with the suffix of \sphinxcode{\sphinxupquote{.qq.png}} is the Q\sphinxhyphen{}Q plots for p\sphinxhyphen{}values of GWAS summary statistics and gene\sphinxhyphen{}based association tests by GATES and ECS.

\sphinxstepscope


\chapter{Options}
\label{\detokenize{options:options}}\label{\detokenize{options:id1}}\label{\detokenize{options::doc}}
\sphinxAtStartPar
The options for {\hyperref[\detokenize{options:option-vcf}]{\sphinxcrossref{\DUrole{std,std-ref}{Reference population genotypes}}}}, {\hyperref[\detokenize{options:option-gwas}]{\sphinxcrossref{\DUrole{std,std-ref}{GWAS summary statistics}}}}, and {\hyperref[\detokenize{options:option-misc}]{\sphinxcrossref{\DUrole{std,std-ref}{Miscellaneous global options}}}} act on all analyses. For clarity, we have categorized the other parameters by {\hyperref[\detokenize{options:option-assoc}]{\sphinxcrossref{\DUrole{std,std-ref}{Gene\sphinxhyphen{}based association and heritability}}}}, {\hyperref[\detokenize{options:option-dese}]{\sphinxcrossref{\DUrole{std,std-ref}{DESE}}}} and {\hyperref[\detokenize{options:option-emic}]{\sphinxcrossref{\DUrole{std,std-ref}{EMIC}}}}, although this has resulted in some duplication of parameters.

\sphinxAtStartPar
In the “Default” columns of the following tables, “null” denotes that the flag works with an argument but there is no default value; “n/a” denotes that the flag works without an argument.


\section{Reference population genotypes}
\label{\detokenize{options:reference-population-genotypes}}\label{\detokenize{options:option-vcf}}
\sphinxAtStartPar
These options work on the VCF file of reference population genotypes. Only SNPs that pass the filters will be used for subsequent analyses. These options act on all analyses.


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{2}{7}|\X{4}{7}|\X{1}{7}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Default
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}vcf\sphinxhyphen{}ref}}
&
\sphinxAtStartPar
Specifies a VCF file of genotypes sampled from a reference population. These genotypes are used to estimate LD correlation coefficients among SNPs. For VCF files of separated chromosomes, use wildcards with quotes like \sphinxcode{\sphinxupquote{"chr*.vcf.gz"}}.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}keep\sphinxhyphen{}ref}}
&
\sphinxAtStartPar
Keep the parsed VCF files as KGGSEE object format in a folder named \sphinxcode{\sphinxupquote{VCFRefhg*}} under the output folder.
&
\sphinxAtStartPar
n/a
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}saved\sphinxhyphen{}ref}}
&
\sphinxAtStartPar
Specifies the folder of genotypes of reference population in KGGSEE object format, which is saved by the \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}keep\sphinxhyphen{}ref}}.  Reading KGGSEE object format files is faster than parsing VCF files.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}maf\sphinxhyphen{}le}}
&
\sphinxAtStartPar
Filter SNPs with a minor allele frequency lower than the setting.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{0.05}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}hwe\sphinxhyphen{}all}}
&
\sphinxAtStartPar
Filter SNPs with a p\sphinxhyphen{}value of rejecting Hardy\sphinxhyphen{}Weinberg equilibrium lower than the setting.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{1E\sphinxhyphen{}5}}
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\newpage
\section{GWAS summary statistics}
\label{\detokenize{options:gwas-summary-statistics}}\label{\detokenize{options:option-gwas}}
\sphinxAtStartPar
These options work on the GWAS summary statistics and act on all analyses.


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{2}{7}|\X{4}{7}|\X{1}{7}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Default
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}sum\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a whitespace delimitated file of GWAS summary statistics.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}chrom\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of chromosomes.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{CHR}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}pos\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of coordinates.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{BP}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}p\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of p\sphinxhyphen{}values.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{P}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}a1\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of the reference allele to calculate effect sizes.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{A1}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}a2\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of the other allele.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{A2}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}freq\sphinxhyphen{}a1\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of the frequency of the allele specified by \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}a1\sphinxhyphen{}col}}.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{FRQ\_U}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}beta\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of effect sizes.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}beta\sphinxhyphen{}type}}
&
\sphinxAtStartPar
Specifies the type of effect sizes:  \sphinxcode{\sphinxupquote{0}} for the linear regression coefficient of a quantitative phenotype; \sphinxcode{\sphinxupquote{1}} for the logarithm of odds ratio or logistic regression coefficient of a qualitative phenotype; \sphinxcode{\sphinxupquote{2}} for an odds ratio of a qualitative phenotype.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}se\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of standard errors of effect sizes. Note: even if the effect size is provided as an odds ratio, this is still the standard error of the logarithm (base e) of the odds ratio.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{SE}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}nmiss\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of sample sizes for a quantitative phenotype.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{Neff}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}case\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of case sample sizes for a qualitative phenotype.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}control\sphinxhyphen{}col}}
&
\sphinxAtStartPar
Specifies the column of control sample sizes for a qualitative phenotype.
&
\sphinxAtStartPar
null
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\section{Gene\sphinxhyphen{}based association and heritability}
\label{\detokenize{options:gene-based-association-and-heritability}}\label{\detokenize{options:option-assoc}}

\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{2}{7}|\X{4}{7}|\X{1}{7}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Default
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}assoc}}
&
\sphinxAtStartPar
Triggers gene\sphinxhyphen{}based association tests.
&
\sphinxAtStartPar
n/a
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}estimate\sphinxhyphen{}heritability}}
&
\sphinxAtStartPar
Triggers gene\sphinxhyphen{}based association tests and estimation of gene heritability.
&
\sphinxAtStartPar
n/a
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}prevalence}}
&
\sphinxAtStartPar
Specifies the proportion of cases in the population when estimating the heritability of a qualitative phenotype.
&
\sphinxAtStartPar
0.01
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}}
&
\sphinxAtStartPar
Specifies the number of basepairs to extend at both ends of a gene, when considering SNPs belonging to the gene.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{5000}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a fasta\sphinxhyphen{}styled file of eQTL summary statistics. If this flag is used, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}} is overridden, and eQTLs of a gene or transcript will be grouped and tested.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}eqtl\sphinxhyphen{}p}}
&
\sphinxAtStartPar
Specifies the threshold of eQTL p\sphinxhyphen{}values. Only eQTLs with a p\sphinxhyphen{}value lower than the threshold will be used. The default is \sphinxcode{\sphinxupquote{0.01}} when performing gene\sphinxhyphen{}based association tests and heritability estimating.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{0.01}}
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\section{DESE}
\label{\detokenize{options:dese}}\label{\detokenize{options:option-dese}}

\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{2}{7}|\X{4}{7}|\X{1}{7}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Default
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}gene\sphinxhyphen{}finemapping}}
&
\sphinxAtStartPar
Trigers the DESE, eDESE or SelDP.
&
\sphinxAtStartPar
n/a
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}expression\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a gene expression file that contains means and standard errors of gene expressions in multiple tissues.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}multiple\sphinxhyphen{}testing}}
&
\sphinxAtStartPar
Specifies the method for multiple testing correction. \sphinxcode{\sphinxupquote{bonf}} denotes performing Bonferroni correction; \sphinxcode{\sphinxupquote{benfdr}} denotes controlling false discovery rate by the Benjamini\textendash{}Hochberg method; \sphinxcode{\sphinxupquote{fixed}} denotes no correction.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{bonf}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}p\sphinxhyphen{}value\sphinxhyphen{}cutoff}}
&
\sphinxAtStartPar
Specifies the threshold of the adjusted p\sphinxhyphen{}value for fine\sphinxhyphen{}mapping. Only genes with an adjusted p\sphinxhyphen{}value lower than the threshold will be retained for fine\sphinxhyphen{}mapping.
&
\sphinxAtStartPar
0.05
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}top\sphinxhyphen{}gene}}
&
\sphinxAtStartPar
Specifies the maximum number of genes with the smallest p\sphinxhyphen{}values that will be retained for fine\sphinxhyphen{}mapping.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}geneset\sphinxhyphen{}db}}
&
\sphinxAtStartPar
Specifies \sphinxhref{http://www.gsea-msigdb.org/gsea/msigdb/index.jsp}{MSigDB} gene sets for enrichment analysis:

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{cura}}: C2. curated gene sets;

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{cgp}}: C2. chemical and genetic perturbations;

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{cano}}: C2. canonical pathways;

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{cmop}}: C4. computational gene sets;

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{onto}}: C5. ontology gene sets;

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{onco}}: C6. oncogenic signature gene sets;

\sphinxAtStartPar
\sphinxcode{\sphinxupquote{immu}}: C7. immunologic signature gene sets.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}geneset\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a user\sphinxhyphen{}defined file of gene sets for enrichment analysis.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}}
&
\sphinxAtStartPar
Specifies the number of basepairs to extend at both ends of a gene, when considering SNPs belonging to the gene.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{5000}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a fasta\sphinxhyphen{}styled file of eQTL summary statistics. If this flag is used, \sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}neargene}} is overridden, and eQTLs of a gene or transcript will be grouped and tested.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}eqtl\sphinxhyphen{}p}}
&
\sphinxAtStartPar
Specifies the threshold of eQTL p\sphinxhyphen{}values. Only eQTLs with a p\sphinxhyphen{}value lower than the threshold will be used. The default is \sphinxcode{\sphinxupquote{0.01}} when performing DESE.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{0.01}}
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\section{EMIC}
\label{\detokenize{options:emic}}\label{\detokenize{options:option-emic}}

\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{2}{7}|\X{4}{7}|\X{1}{7}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Default
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic}}
&
\sphinxAtStartPar
Triggers the EMIC.
&
\sphinxAtStartPar
n/a
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}eqtl\sphinxhyphen{}file}}
&
\sphinxAtStartPar
Specifies a fasta\sphinxhyphen{}styled file of eQTL summary statistics.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}filter\sphinxhyphen{}eqtl\sphinxhyphen{}p}}
&
\sphinxAtStartPar
Specifies the threshold of eQTL p\sphinxhyphen{}values. Only eQTLs with a p\sphinxhyphen{}value lower than the threshold will be used. The default is \sphinxcode{\sphinxupquote{1E\sphinxhyphen{}4}} when performing EMIC.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{1E\sphinxhyphen{}4}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}ld\sphinxhyphen{}pruning\sphinxhyphen{}mr}}
&
\sphinxAtStartPar
Specifies the threshold of LD coefficient when pruning variants. For each gene or transcript, eQTLs with LD coefficients higher than the threshold will be pruned.
&
\sphinxAtStartPar
0.5
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic\sphinxhyphen{}pfm\sphinxhyphen{}p}}
&
\sphinxAtStartPar
Specifies the p\sphinxhyphen{}value threshold to further perform an EMIC pleiotropy fine\sphinxhyphen{}mapping (EMIC\sphinxhyphen{}PFM) analysis. If the EMIC p\sphinxhyphen{}value of a gene is lower than the threshold, an EMIC\sphinxhyphen{}PFM will be performed to control the false\sphinxhyphen{}positive caused by pleiotropy.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{2.5E\sphinxhyphen{}6}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}emic\sphinxhyphen{}plot\sphinxhyphen{}p}}
&
\sphinxAtStartPar
Specifies the p\sphinxhyphen{}value threshold for plotting a scatter plot. Genes with an EMIC p\sphinxhyphen{}value lower than the threshold will be plotted.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{2.5E\sphinxhyphen{}3}}
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\section{Miscellaneous global options}
\label{\detokenize{options:miscellaneous-global-options}}\label{\detokenize{options:option-misc}}
\sphinxAtStartPar
These options act on all analyses.


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabular}[t]{|\X{2}{9}|\X{5}{9}|\X{2}{9}|}
\hline
\sphinxstyletheadfamily 
\sphinxAtStartPar
Flag
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Description
&\sphinxstyletheadfamily 
\sphinxAtStartPar
Default
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}nt}}
&
\sphinxAtStartPar
Specifies the number of threads.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{4}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}buildver}}
&
\sphinxAtStartPar
Specifies the reference genome version of the coordinates. The supported versions are \sphinxcode{\sphinxupquote{hg19}} and \sphinxcode{\sphinxupquote{hg38}}.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{hg19}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}db\sphinxhyphen{}gene}}
&
\sphinxAtStartPar
Specifies the database of gene annotations. \sphinxcode{\sphinxupquote{refgene}} for RefSeq Genes; \sphinxcode{\sphinxupquote{gencode}} for GENCODE; \sphinxcode{\sphinxupquote{refgene,gencode}} for both.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{gencode}}
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}excel}}
&
\sphinxAtStartPar
Output results in Excel format.
&
\sphinxAtStartPar
n/a
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}only\sphinxhyphen{}hgnc\sphinxhyphen{}gene}}
&
\sphinxAtStartPar
Only genes with an HGNC\sphinxhyphen{}approved gene symbol are considered in analyses.
&
\sphinxAtStartPar
n/a
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}out}}
&
\sphinxAtStartPar
Specifies the output prefix of results.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}regions\sphinxhyphen{}bed}}
&
\sphinxAtStartPar
Specifies a \sphinxhref{https://en.wikipedia.org/wiki/BED\_(file\_format)}{BED file} to define customized gene coordinates instead of the annotation from RefSeqGene or GENCODE. The first three columns of the BED file define gene coordinate and are mandatory; the fourth column defines gene names and is optional. When the fourth column is absent, a gene name of the format like \sphinxcode{\sphinxupquote{chr1:100\sphinxhyphen{}200}} will be allocated.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}regions\sphinxhyphen{}out}}
&
\sphinxAtStartPar
Specifies genomic regions to be excluded in analyses, e.g. \sphinxcode{\sphinxupquote{chr1,chr2:2323\sphinxhyphen{}34434,chr2:43455\sphinxhyphen{}345555}}.
&
\sphinxAtStartPar
null
\\
\hline
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{\sphinxhyphen{}\sphinxhyphen{}resource}}
&
\sphinxAtStartPar
Specifies the path KGGSEE running resource data.
&
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{resources/}} under the folder of \sphinxcode{\sphinxupquote{kggsee.jar}}
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}



\renewcommand{\indexname}{Index}
\printindex
\end{document}
